// quasar/lib/array-fill.js
"use strict";

require("earlgrey-runtime/5");var $targ$12 = undefined;var util$0 = undefined;var identify$0 = undefined;var fork__minus__identify$0 = undefined;var tuck$0 = undefined;var next__minus__id$0 = undefined;var ID$0 = undefined;var methods$0 = undefined;var longest__minus__run$0 = undefined;var proxy$0 = undefined;var deprox$0 = undefined;var $targ$0 = undefined;var $targ$1 = undefined;var array__minus__clobber__minus__helper$0 = undefined;var $targ$2 = undefined;var $targ$3 = undefined;var $targ$4 = undefined;var $targ$5 = undefined;util$0 = require("./util");identify$0 = getProperty(util$0, "identify", "util");fork__minus__identify$0 = getProperty(util$0, "forkIdentify", "util");tuck$0 = getProperty(util$0, "tuck", "util");next__minus__id$0 = getProperty(util$0, "nextId", "util");ID$0 = getProperty(util$0, "ID", "util");methods$0 = getProperty(util$0, "methods", "util");longest__minus__run$0 = getProperty(util$0, "longestRun", "util");proxy$0 = require("./proxy");deprox$0 = getProperty(proxy$0, "deprox", "proxy");$targ$0 = function () {
  var t0$0 = undefined;var $targ$6 = undefined;var __at___$0 = undefined;var self$0 = undefined;$targ$6 = this;t0$0 = $targ$6;__at___$0 = t0$0;self$0 = t0$0;[__at___$0, self$0];return __at___$0.slice(0);
};Array.prototype[methods$0.fork] = $targ$0;$targ$1 = function () {
  var tmp$0 = undefined;var m$1 = undefined;var arr$0 = undefined;var splices$0 = undefined;var p$0 = undefined;var dir$0 = undefined;var wrapper$0 = undefined;var t0$1 = undefined;var m$0$0 = undefined;m$0$0 = arguments;t0$1 = m$0$0.length;if (t0$1 >= 2 && t0$1 <= 3) {
    p$0 = m$0$0[0];dir$0 = m$0$0[1];if (2 >= t0$1) {
      wrapper$0 = this;
    } else {
      wrapper$0 = m$0$0[2];
    }arr$0 = this;identify$0(wrapper$0);tmp$0 = p$0.getFor(wrapper$0);if (tmp$0 === undefined || tmp$0 === null) {
      splices$0 = tmp$0;
    } else {
      splices$0 = tmp$0["@splice"];
    }if (!splices$0) {
      return null;
    }m$1 = null;$0: for (var _iterator = enumerate(splices$0).reverse()[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      var _iterator2, _step2;

      var _iterator3, _step3;

      (function () {
        m$1 = _step.value;
        var m$2 = undefined;var acc$0 = undefined;var temp$0 = undefined;var m$3 = undefined;var acc$1 = undefined;var temp$1 = undefined;var m$4 = undefined;var previous$0 = undefined;var acq$0 = undefined;var n$0 = undefined;var repl$0 = undefined;var t0$3 = undefined;var t1$1 = undefined;var i$0 = undefined;var ph$0$0 = undefined;var t0$2 = undefined;var t1$0 = undefined;t0$2 = m$1;if (Array.isArray(t0$2) && (t1$0 = t0$2.length, t1$0 === 2)) {
          i$0 = t0$2[0];ph$0$0 = t0$2[1];t0$3 = ph$0$0;if (Array.isArray(t0$3) && (t1$1 = t0$3.length, t1$1 === 2)) {
            n$0 = t0$3[0];repl$0 = t0$3[1];previous$0 = arr$0.splice(i$0, n$0);acc$0 = [];temp$0 = null;m$2 = null;$2: for (_iterator2 = zip(repl$0.slice(0, previous$0.length), previous$0)[Symbol.iterator](); !(_step2 = _iterator2.next()).done;) {
              m$2 = _step2.value;
              var newv$0 = undefined;var oldv$0 = undefined;var t0$4 = undefined;var t1$2 = undefined;t0$4 = m$2;if (Array.isArray(t0$4) && (t1$2 = t0$4.length, t1$2 === 2)) {
                newv$0 = t0$4[0];oldv$0 = t0$4[1];temp$0 = dir$0.replace(oldv$0, deprox$0(newv$0));acc$0.push(temp$0);
              } else {
                ___match_error(m$2);
              }
            }acq$0 = acc$0;acq$0 = acq$0.concat((acc$1 = [], temp$1 = null, m$3 = null, (function () {
              $3: for (var _iterator3 = repl$0.slice(previous$0.length)[Symbol.iterator](), _step3; !(_step3 = _iterator3.next()).done;) {
                m$3 = _step3.value;
                var newv$1 = undefined;newv$1 = m$3;temp$1 = dir$0.acquire(deprox$0(newv$1));acc$1.push(temp$1);
              }
            })(), acc$1));m$4 = null;$1: for (_iterator3 = previous$0.slice(repl$0.length)[Symbol.iterator](); !(_step3 = _iterator3.next()).done;) {
              m$4 = _step3.value;
              var oldv$1 = undefined;oldv$1 = m$4;dir$0.exit(oldv$1);
            }send(send(arr$0, "splice", true), [i$0, 0].concat(acq$0));
          } else {
            undefined;
          }
        } else {
          ___match_error(m$1);
        }
      })();
    }return undefined;
  } else {
    return ___match_error(m$0$0, "{p, dir, wrapper = this}");
  }
};Array.prototype[methods$0.patch] = $targ$1;array__minus__clobber__minus__helper$0 = function arrayClobberHelper() {
  var m$6 = undefined;var m$7 = undefined;var acc$2 = undefined;var temp$2 = undefined;var cmp$0 = undefined;var m$8 = undefined;var m$9 = undefined;var m$10 = undefined;var acc$3 = undefined;var temp$5 = undefined;var m$11 = undefined;var idpos$0 = undefined;var indices$0 = undefined;var run$0 = undefined;var newids$0 = undefined;var disposable$0 = undefined;var old__minus__current$0 = undefined;var new__minus__current$0 = undefined;var operations$0 = undefined;var arr$1 = undefined;var new__minus__arr$0 = undefined;var duplicate$0 = undefined;var t0$5 = undefined;var m$5$0 = undefined;m$5$0 = arguments;t0$5 = m$5$0.length;if (t0$5 >= 2 && t0$5 <= 3) {
    arr$1 = m$5$0[0];new__minus__arr$0 = m$5$0[1];if (2 >= t0$5) {
      duplicate$0 = true;
    } else {
      duplicate$0 = m$5$0[2];
    }idpos$0 = {};m$6 = null;$4: for (var _iterator = enumerate(arr$1)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      m$6 = _step.value;
      var $targ$7 = undefined;var id$0 = undefined;var t0$7 = undefined;var t1$4 = undefined;var t2$1 = undefined;var i$1 = undefined;var x$0 = undefined;var ph$1$0 = undefined;var t0$6 = undefined;var t1$3 = undefined;var t2$0 = undefined;t0$6 = m$6;if (Array.isArray(t0$6) && (t1$3 = t0$6.length, t1$3 === 2)) {
        i$1 = t0$6[0];t2$0 = t0$6[1];x$0 = t2$0;ph$1$0 = t2$0;t0$7 = ph$1$0;t1$4 = ID$0;if (___hasprop(t0$7, t1$4) && (t2$1 = send(t0$7, t1$4), typeof t2$1 === "number")) {
          id$0 = t2$1;$targ$7 = [i$1, x$0];idpos$0[id$0] = $targ$7;void 0;
        } else {
          continue $4;
        }
      } else {
        ___match_error(m$6);
      }
    }acc$2 = [];temp$2 = null;m$7 = null;$8: for (var _iterator2 = enumerate(new__minus__arr$0)[Symbol.iterator](), _step2; !(_step2 = _iterator2.next()).done;) {
      m$7 = _step2.value;
      var id$1 = undefined;var t0$9 = undefined;var t1$6 = undefined;var t2$3 = undefined;var i$2 = undefined;var x$1 = undefined;var ph$2$0 = undefined;var t0$8 = undefined;var t1$5 = undefined;var t2$2 = undefined;t0$8 = m$7;if (Array.isArray(t0$8) && (t1$5 = t0$8.length, t1$5 === 2)) {
        i$2 = t0$8[0];t2$2 = t0$8[1];x$1 = t2$2;ph$2$0 = t2$2;t0$9 = ph$2$0;t1$6 = ID$0;if (___hasprop(t0$9, t1$6) && (t2$3 = send(t0$9, t1$6), typeof t2$3 === "number" && (id$1 = t2$3, send(idpos$0, id$1)))) {
          temp$2 = [i$2, send(idpos$0, id$1)[0]];
        } else {
          continue $8;
        }acc$2.push(temp$2);
      } else {
        ___match_error(m$7);
      }
    }indices$0 = acc$2;cmp$0 = function cmp(temp$3$0, temp$4$0) {
      var t0$10 = undefined;var t1$7 = undefined;var t0$11 = undefined;var t1$8 = undefined;var p1$0 = undefined;var p2$0 = undefined;t0$10 = temp$3$0;if (Array.isArray(t0$10) && (t1$7 = t0$10.length, t1$7 === 2)) {
        t0$10[0];p1$0 = t0$10[1];
      } else {
        ___match_error(temp$3$0);
      }t0$11 = temp$4$0;if (Array.isArray(t0$11) && (t1$8 = t0$11.length, t1$8 === 2)) {
        t0$11[0];p2$0 = t0$11[1];
      } else {
        ___match_error(temp$4$0);
      }return p1$0 < p2$0;
    };run$0 = longest__minus__run$0(indices$0, cmp$0, [-1, -1]);newids$0 = {};m$8 = null;$5: for (var _iterator3 = new__minus__arr$0[Symbol.iterator](), _step3; !(_step3 = _iterator3.next()).done;) {
      m$8 = _step3.value;
      var $targ$8 = undefined;var x$2 = undefined;var id$2 = undefined;var t0$12 = undefined;var t1$9 = undefined;var t2$4 = undefined;t0$12 = m$8;x$2 = t0$12;t1$9 = ID$0;if (___hasprop(t0$12, t1$9) && (t2$4 = send(t0$12, t1$9), typeof t2$4 === "number" && (id$2 = t2$4, true))) {
        $targ$8 = true;newids$0[id$2] = $targ$8;void 0;
      } else {
        false;
      }
    }disposable$0 = {};m$9 = null;$6: for (var _iterator4 = arr$1[Symbol.iterator](), _step4; !(_step4 = _iterator4.next()).done;) {
      m$9 = _step4.value;
      var $targ$9 = undefined;var x$3 = undefined;var id$3 = undefined;var t0$13 = undefined;var t1$10 = undefined;var t2$5 = undefined;t0$13 = m$9;x$3 = t0$13;t1$10 = ID$0;if (___hasprop(t0$13, t1$10) && (t2$5 = send(t0$13, t1$10), typeof t2$5 === "number" && (id$3 = t2$5, !send(newids$0, id$3)))) {
        $targ$9 = true;disposable$0[id$3] = $targ$9;void 0;
      } else {
        false;
      }
    }if (duplicate$0) {
      acc$3 = [];temp$5 = null;m$10 = null;$9: for (var _iterator5 = run$0[Symbol.iterator](), _step5; !(_step5 = _iterator5.next()).done;) {
        m$10 = _step5.value;
        var new__minus__i$0 = undefined;var old__minus__i$0 = undefined;var t0$14 = undefined;var t1$11 = undefined;t0$14 = m$10;if (Array.isArray(t0$14) && (t1$11 = t0$14.length, t1$11 === 2)) {
          new__minus__i$0 = t0$14[0];old__minus__i$0 = t0$14[1];if (send(arr$1, old__minus__i$0) && send(send(arr$1, old__minus__i$0, true), ID$0)) {
            delete idpos$0[send(send(arr$1, old__minus__i$0, true), ID$0)];
          }acc$3.push(temp$5);
        } else {
          ___match_error(m$10);
        }
      }acc$3;
    }run$0.push([new__minus__arr$0.length, arr$1.length]);old__minus__current$0 = 0;new__minus__current$0 = 0;operations$0 = [];m$11 = null;$7: for (var _iterator6 = run$0[Symbol.iterator](), _step6; !(_step6 = _iterator6.next()).done;) {
      m$11 = _step6.value;
      var m$13 = undefined;var acc$4 = undefined;var temp$6 = undefined;var n$1 = undefined;var fresh$0 = undefined;var inserts$0 = undefined;var new__minus__i$1 = undefined;var old__minus__i$1 = undefined;var t0$15 = undefined;var t1$12 = undefined;t0$15 = m$11;if (Array.isArray(t0$15) && (t1$12 = t0$15.length, t1$12 === 2)) {
        new__minus__i$1 = t0$15[0];old__minus__i$1 = t0$15[1];n$1 = old__minus__i$1 - old__minus__current$0;fresh$0 = new__minus__arr$0.slice(new__minus__current$0, new__minus__i$1);$10: while (n$1 > 0 && old__minus__current$0 < old__minus__i$1 && fresh$0.length > 0) {
          var bridge$$560$0 = undefined;var bridge$$559$0 = undefined;var bridge$$558$0 = undefined;var bridge$$557$0 = undefined;var id$4 = undefined;var bridge$$556$0 = undefined;var t0$16 = undefined;var t1$13 = undefined;var m$12$0 = undefined;m$12$0 = send(arr$1, old__minus__current$0);bridge$$556$0 = m$12$0;if ((bridge$$557$0 = bridge$$556$0, (bridge$$558$0 = bridge$$557$0, (bridge$$559$0 = bridge$$558$0, (bridge$$560$0 = bridge$$559$0, bridge$$560$0 === null || bridge$$560$0 === void 0) || typeof bridge$$559$0 === "string") || typeof bridge$$558$0 === "number") || typeof bridge$$557$0 === "boolean") || typeof bridge$$556$0 === "function") {
            operations$0.push(["clobber", new__minus__current$0, fresh$0.shift()]);old__minus__current$0 = old__minus__current$0 + 1;new__minus__current$0 = new__minus__current$0 + 1;n$1 = n$1 - 1;
          } else {
            t0$16 = ID$0;if (___hasprop(m$12$0, t0$16) && (t1$13 = send(m$12$0, t0$16), typeof t1$13 === "number" && (id$4 = t1$13, send(disposable$0, id$4)))) {
              operations$0.push(["clobber", new__minus__current$0, fresh$0.shift()]);old__minus__current$0 = old__minus__current$0 + 1;new__minus__current$0 = new__minus__current$0 + 1;n$1 = n$1 - 1;
            } else {
              break $10;
            }
          }
        }acc$4 = [];temp$6 = null;m$13 = null;$11: for (var _iterator7 = fresh$0[Symbol.iterator](), _step7; !(_step7 = _iterator7.next()).done;) {
          m$13 = _step7.value;
          var entry$1 = undefined;var entry$0 = undefined;var orig$0 = undefined;var t0$17 = undefined;var t1$14 = undefined;var t2$6 = undefined;t0$17 = m$13;entry$0 = t0$17;t1$14 = send(idpos$0, send(entry$0, ID$0) || -1);if (Array.isArray(t1$14) && (t2$6 = t1$14.length, t2$6 === 2 && (t1$14[0], orig$0 = t1$14[1], entry$0 && duplicate$0))) {
            delete idpos$0[send(entry$0, ID$0)];temp$6 = ["keep", orig$0];acc$4.push(temp$6);
          } else {
            entry$1 = m$13;temp$6 = ["acquire", entry$1];acc$4.push(temp$6);
          }
        }inserts$0 = acc$4;operations$0.push(["splice", new__minus__current$0, n$1, inserts$0]);old__minus__current$0 = old__minus__i$1 + 1;new__minus__current$0 = new__minus__i$1 + 1;new__minus__current$0;
      } else {
        ___match_error(m$11);
      }
    }return operations$0;
  } else {
    return ___match_error(m$5$0, "{arr, new-arr, duplicate = true}");
  }
};$targ$2 = function (new__minus__arr$1, dir$1) {
  var m$14 = undefined;var m$17 = undefined;var arr$2 = undefined;var exits$0 = undefined;var keeps$0 = undefined;arr$2 = this;if (!Array.isArray(new__minus__arr$1)) {
    return dir$1.replace(arr$2, new__minus__arr$1, false);
  }dir$1.unlist(arr$2);exits$0 = new Set();keeps$0 = new Set();m$14 = null;$12: for (var _iterator = array__minus__clobber__minus__helper$0(arr$2, new__minus__arr$1, dir$1.duplicate)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$14 = _step.value;
    var $targ$10 = undefined;var m$15 = undefined;var m$16 = undefined;var acc$5 = undefined;var temp$7 = undefined;var acquired$0 = undefined;var start$0 = undefined;var n$2 = undefined;var inserts$1 = undefined;var i$3 = undefined;var value$0 = undefined;var $$692$0 = undefined;var $$693$0 = undefined;var $$694$0 = undefined;var t0$18 = undefined;var t1$15 = undefined;t0$18 = m$14;if (($$693$0 = Array.isArray(t0$18)) && (t1$15 = t0$18.length, t1$15 === 3 && t0$18[0] === "clobber")) {
      i$3 = t0$18[1];value$0 = t0$18[2];$targ$10 = dir$1.clobber(send(arr$2, i$3), value$0);arr$2[i$3] = $targ$10;void 0;
    } else {
      if ($$693$0 && (t1$15 === 4 && t0$18[0] === "splice")) {
        start$0 = t0$18[1];n$2 = t0$18[2];inserts$1 = t0$18[3];m$15 = null;$14: for (var _iterator2 = range(start$0, start$0 + n$2)[Symbol.iterator](), _step2; !(_step2 = _iterator2.next()).done;) {
          m$15 = _step2.value;
          var i$4 = undefined;i$4 = m$15;exits$0.add(send(arr$2, i$4));
        }acc$5 = [];temp$7 = null;m$16 = null;$15: for (var _iterator3 = inserts$1[Symbol.iterator](), _step3; !(_step3 = _iterator3.next()).done;) {
          m$16 = _step3.value;
          var x$5 = undefined;var x$4 = undefined;var $$750$0 = undefined;var $$751$0 = undefined;var $$752$0 = undefined;var $$753$0 = undefined;var t0$19 = undefined;var t1$16 = undefined;t0$19 = m$16;if (($$751$0 = Array.isArray(t0$19)) && (t1$16 = t0$19.length, ($$753$0 = t1$16 === 2) && t0$19[0] === "keep")) {
            x$4 = t0$19[1];keeps$0.add(x$4);temp$7 = x$4;acc$5.push(temp$7);
          } else {
            if ($$753$0 && t0$19[0] === "acquire") {
              x$5 = t0$19[1];temp$7 = dir$1.acquire(x$5);acc$5.push(temp$7);
            } else {
              ___match_error(m$16);
            }
          }
        }acquired$0 = acc$5;send(send(arr$2, "splice", true), [start$0, n$2].concat(acquired$0));
      } else {
        ___match_error(m$14);
      }
    }
  }m$17 = null;$13: for (var _iterator4 = exits$0[Symbol.iterator](), _step4; !(_step4 = _iterator4.next()).done;) {
    m$17 = _step4.value;
    var x$6 = undefined;x$6 = m$17;if (!keeps$0.has(x$6)) {
      dir$1.exit(x$6);
    } else {
      false;
    }
  }tuck$0(arr$2, ID$0, send(new__minus__arr$1, ID$0));dir$1.list(arr$2);return arr$2;
};Array.prototype[methods$0.clobber] = $targ$2;$targ$3 = function (i$5, value$1) {
  var $targ$11 = undefined;$targ$11 = value$1;this[i$5] = $targ$11;return void 0;
};Array.prototype[methods$0.set] = $targ$3;$targ$4 = function () {
  var m$19 = undefined;var acc$6 = undefined;var temp$8 = undefined;var arr$3 = undefined;var id$5 = undefined;var wrapper$1 = undefined;var t0$20 = undefined;var m$18$0 = undefined;m$18$0 = arguments;t0$20 = m$18$0.length;if (t0$20 >= 0 && t0$20 <= 1) {
    if (0 >= t0$20) {
      wrapper$1 = this;
    } else {
      wrapper$1 = m$18$0[0];
    }arr$3 = this;id$5 = send(wrapper$1, ID$0);acc$6 = [];temp$8 = null;m$19 = null;$16: for (var _iterator = enumerate(arr$3)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      (function () {
        m$19 = _step.value;
        var setter$0 = undefined;var i$6 = undefined;var v$0 = undefined;var t0$21 = undefined;var t1$17 = undefined;t0$21 = m$19;if (Array.isArray(t0$21) && (t1$17 = t0$21.length, t1$17 === 2)) {
          i$6 = t0$21[0];v$0 = t0$21[1];setter$0 = function setter(new__minus__v$0) {
            return send(arr$3, methods$0.set, true)(i$6, new__minus__v$0);
          };temp$8 = [i$6, v$0, setter$0];acc$6.push(temp$8);
        } else {
          ___match_error(m$19);
        }
      })();
    }return acc$6;
  } else {
    return ___match_error(m$18$0, "{wrapper = this}");
  }
};Array.prototype[methods$0.iterateProperties] = $targ$4;$targ$5 = regeneratorRuntime.mark(function callee$0$0(reads$0, writes$0) {
  var m$21, acc$7, temp$9, m$22, acc$8, temp$10, mask$0, taints$0, splices$1, m$20$0, _iterator, _step, t0$22, t1$18, t2$7, t3$0, _iterator2, _step2, tmp$1, n$3, v$2, v$1, $$923$0, $$924$0, $$925$0, $$926$0, t0$23, t1$19;

  return regeneratorRuntime.wrap(function callee$0$0$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        m$21 = undefined;
        acc$7 = undefined;
        temp$9 = undefined;
        m$22 = undefined;
        acc$8 = undefined;
        temp$10 = undefined;
        mask$0 = undefined;
        taints$0 = undefined;
        splices$1 = undefined;
        m$20$0 = undefined;
        m$20$0 = writes$0["@splice"];
        if (!(m$20$0 === void 0)) {
          context$1$0.next = 15;
          break;
        }

        return context$1$0.abrupt("return", undefined);

      case 15:
        splices$1 = m$20$0;mask$0 = 3;acc$7 = [];temp$9 = null;m$21 = null;$18: for (_iterator = splices$1[Symbol.iterator](); !(_step = _iterator.next()).done;) {
          m$21 = _step.value;
          t0$22 = undefined;
          t1$18 = undefined;
          t2$7 = undefined;
          t3$0 = undefined;
          if (m$21 === void 0) {
            temp$9 = 3 & mask$0;acc$7.push(temp$9);
          } else {
            t0$22 = m$21;if (Array.isArray(t0$22) && (t1$18 = t0$22.length, t1$18 === 2 && (t0$22[0] === 1 && (t2$7 = t0$22[1], Array.isArray(t2$7) && (t3$0 = t2$7.length, t3$0 === 1))))) {
              t2$7[0];temp$9 = 1 & mask$0;acc$7.push(temp$9);
            } else {
              mask$0 = 0;temp$9 = 0;acc$7.push(temp$9);
            }
          }
        }taints$0 = acc$7;acc$8 = [];temp$10 = null;m$22 = null;_iterator2 = items(reads$0)[Symbol.iterator]();

      case 26:
        if ((_step2 = _iterator2.next()).done) {
          context$1$0.next = 58;
          break;
        }

        m$22 = _step2.value;
        tmp$1 = undefined;
        n$3 = undefined;
        v$2 = undefined;
        v$1 = undefined;
        $$923$0 = undefined;
        $$924$0 = undefined;
        $$925$0 = undefined;
        $$926$0 = undefined;
        t0$23 = undefined;
        t1$19 = undefined;
        t0$23 = m$22;
        if (!(($$924$0 = Array.isArray(t0$23)) && (t1$19 = t0$23.length, ($$926$0 = t1$19 === 2) && t0$23[0] === "@iter"))) {
          context$1$0.next = 47;
          break;
        }

        v$1 = t0$23[1];context$1$0.next = 43;
        return v$1;

      case 43:
        temp$10 = context$1$0.sent;
        acc$8.push(temp$10);context$1$0.next = 56;
        break;

      case 47:
        if (!($$926$0 && (n$3 = t0$23[0], v$2 = t0$23[1], tmp$1 = v$2, tmp$1 === undefined || tmp$1 === null ? tmp$1 : tmp$1.notEmpty()))) {
          context$1$0.next = 55;
          break;
        }

        if (!nequal(send(taints$0, n$3), 3)) {
          context$1$0.next = 52;
          break;
        }

        context$1$0.next = 51;
        return v$2;

      case 51:
        temp$10 = context$1$0.sent;

      case 52:
        acc$8.push(temp$10);context$1$0.next = 56;
        break;

      case 55:
        false;

      case 56:
        context$1$0.next = 26;
        break;

      case 58:
        return context$1$0.abrupt("return", acc$8);

      case 59:
      case "end":
        return context$1$0.stop();
    }
  }, callee$0$0, this);
});Array.prototype[methods$0.relevantChanges] = $targ$5;$targ$12 = array__minus__clobber__minus__helper$0;exports.arrayClobberHelper = $targ$12;void 0;
