// quasar/lib/cache.js
"use strict";

require("earlgrey-runtime/5");var $targ$2 = undefined;var accum$0 = undefined;var $targ$3 = undefined;var accum$1 = undefined;var $targ$5 = undefined;var accum$2 = undefined;var $targ$6 = undefined;var accum$3 = undefined;var $targ$12 = undefined;var accum$4 = undefined;var $targ$13 = undefined;var accum$5 = undefined;var $targ$18 = undefined;var $targ$19 = undefined;var $targ$20 = undefined;var $targ$21 = undefined;var $targ$22 = undefined;var $targ$23 = undefined;var $targ$24 = undefined;var $targ$25 = undefined;var $targ$26 = undefined;var directory$0 = undefined;var Directory$0 = undefined;var $0$0 = undefined;var proxy$0 = undefined;var deprox$0 = undefined;var reprox$0 = undefined;var relevant$0 = undefined;var intersect$0 = undefined;var util$0 = undefined;var identify$0 = undefined;var fork__minus__identify$0 = undefined;var tuck$0 = undefined;var next__minus__id$0 = undefined;var symbols$0 = undefined;var ID$0 = undefined;var FORKID$0 = undefined;var $1$0 = undefined;var Tracks$0 = undefined;var Counter$0 = undefined;var Chirurgies$0 = undefined;var Patch$0 = undefined;var kaiser$0 = undefined;var arg__minus__key$0 = undefined;var args__minus__key$0 = undefined;var get__minus__replacement$0 = undefined;var CachedResult$0 = undefined;var frame__minus__stack$0 = undefined;var current__minus__frame$0 = undefined;var current__minus__context$0 = undefined;var side__minus__effect$0 = undefined;var emit$0 = undefined;var with__minus__frame$0 = undefined;var Frame$0 = undefined;var _reprox$0 = undefined;var Reactor$0 = undefined;var cached__minus__function$0 = undefined;directory$0 = require("./directory");Directory$0 = getProperty(directory$0, "Directory", "directory");$0$0 = require("./proxy");proxy$0 = getProperty($0$0, "proxy", "./proxy");deprox$0 = getProperty($0$0, "deprox", "./proxy");reprox$0 = getProperty($0$0, "reprox", "./proxy");relevant$0 = require("./relevant");intersect$0 = getProperty(relevant$0, "intersect", "relevant");util$0 = require("./util");identify$0 = getProperty(util$0, "identify", "util");fork__minus__identify$0 = getProperty(util$0, "forkIdentify", "util");tuck$0 = getProperty(util$0, "tuck", "util");next__minus__id$0 = getProperty(util$0, "nextId", "util");symbols$0 = getProperty(util$0, "symbols", "util");ID$0 = getProperty(util$0, "ID", "util");FORKID$0 = getProperty(util$0, "FORKID", "util");$1$0 = require("./tracks");Tracks$0 = getProperty($1$0, "Tracks", "./tracks");Counter$0 = getProperty($1$0, "Counter", "./tracks");Chirurgies$0 = getProperty($1$0, "Chirurgies", "./tracks");Patch$0 = getProperty($1$0, "Patch", "./tracks");kaiser$0 = require("kaiser");arg__minus__key$0 = function argKey(temp$0$0) {
  var t0$0 = undefined;var bridge$$115$0 = undefined;var bridge$$114$0 = undefined;var x$1 = undefined;var bridge$$113$0 = undefined;var t0$1 = undefined;var t1$0 = undefined;var t2$0 = undefined;var t3$0 = undefined;var x$0 = undefined;var ph$0$0 = undefined;t0$0 = temp$0$0;x$0 = t0$0;ph$0$0 = t0$0;if (typeof ph$0$0 === "string") {
    return "S" + x$0;
  } else {
    bridge$$113$0 = ph$0$0;if ((bridge$$114$0 = bridge$$113$0, (bridge$$115$0 = bridge$$114$0, bridge$$115$0 === null || bridge$$115$0 === void 0) || typeof bridge$$114$0 === "number") || typeof bridge$$113$0 === "boolean") {
      return "P" + String(x$0);
    } else {
      t0$1 = ph$0$0;t1$0 = symbols$0.notInspectable;if (___hasprop(t0$1, t1$0) && ((send(t0$1, t1$0) ? true : false) && (t2$0 = getProjector(identify$0)(t0$1), t2$0[0] && (t2$0[1], t3$0 = getProjector(fork__minus__identify$0)(t0$1), t3$0[0])))) {
        t3$0[1];return "O" + String(send(x$0, ID$0)) + ":" + String(send(x$0, FORKID$0));
      } else {
        t0$1 = getProjector(deprox$0)(ph$0$0);if (t0$1[0] && (t1$0 = t0$1[1], x$1 = t1$0, t2$0 = ID$0, ___hasprop(t1$0, t2$0) && typeof send(t1$0, t2$0) === "number")) {
          fork__minus__identify$0(x$1);return "O" + String(send(x$1, ID$0)) + ":" + String(send(x$1, FORKID$0));
        } else {
          return "K" + kaiser$0.serialize(x$0);
        }
      }
    }
  }
};args__minus__key$0 = function argsKey(args$0) {
  return args$0.map(arg__minus__key$0).join("\u0000");
};get__minus__replacement$0 = function getReplacement(temp$1$0) {
  var t0$2 = undefined;var replacement$0 = undefined;var t0$3 = undefined;var cached$0 = undefined;var ph$1$0 = undefined;t0$2 = temp$1$0;cached$0 = t0$2;ph$1$0 = t0$2;t0$3 = ph$1$0;if (___hasprop(t0$3, "replacement")) {
    replacement$0 = t0$3.replacement;return get__minus__replacement$0(replacement$0);
  } else {
    return cached$0;
  }
};CachedResult$0 = function CachedResult(temp$2$0, temp$3$0, temp$4$0, temp$5$0, temp$6$0, temp$7$0, temp$8$0) {
  var __at___$0 = undefined;if (!getChecker(CachedResult$0)(this)) {
    __at___$0 = Object.create(CachedResult$0.prototype);
  } else {
    __at___$0 = this;
  }__at___$0.reactor = temp$2$0;__at___$0.self = temp$3$0;__at___$0.args = temp$4$0;__at___$0.value = temp$5$0;__at___$0.tracks = temp$6$0;__at___$0.effects = temp$7$0;__at___$0.version = temp$8$0;return __at___$0;
};CachedResult$0.prototype.update = function update() {
  var __at___$1 = undefined;var self$0 = undefined;__at___$1 = this;self$0 = this;return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
    var t0$4, t1$1, x$2, p$0, m$1$0, t0$5, $targ$1, $get$0$0, m$2, $$293$0, bridge$$290$0, m$0$0, $targ$0, patch$0, out__minus__patch$0, dir$0, version$0, _iterator, _step, effect$0;

    return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
      while (1) switch (context$2$0.prev = context$2$0.next) {
        case 0:
          t0$4 = undefined;
          t1$1 = undefined;
          x$2 = undefined;
          p$0 = undefined;
          m$1$0 = undefined;
          t0$5 = undefined;
          $targ$1 = undefined;
          $get$0$0 = undefined;
          m$2 = undefined;
          $$293$0 = undefined;
          bridge$$290$0 = undefined;
          m$0$0 = undefined;
          $targ$0 = undefined;
          patch$0 = undefined;
          out__minus__patch$0 = undefined;
          dir$0 = undefined;
          version$0 = undefined;
          $targ$0 = current__minus__context$0();t0$4 = $targ$0;if (___hasprop(t0$4, "patch") && (patch$0 = t0$4.patch, ___hasprop(t0$4, "outPatch") && (out__minus__patch$0 = t0$4.outPatch, ___hasprop(t0$4, "dir") && (t1$1 = t0$4.dir, dir$0 = t1$1, ___hasprop(t1$1, "version"))))) {
            version$0 = t1$1.version;
          } else {
            ___match_error($targ$0, "{\n         => patch\n         => out-patch\n         dir => let dir and {=> version}\n      }");
          }[patch$0, out__minus__patch$0, dir$0, version$0];
          if (patch$0) {
            context$2$0.next = 23;
            break;
          }

          return context$2$0.abrupt("return", __at___$1);

        case 23:
          m$0$0 = version$0;bridge$$290$0 = m$0$0;
          if (!(equal(bridge$$290$0, __at___$1.version) || !__at___$1.version)) {
            context$2$0.next = 29;
            break;
          }

          return context$2$0.abrupt("return", __at___$1);

        case 29:
          if (!(__at___$1.tracks && intersect$0(__at___$1.tracks, patch$0, dir$0))) {
            context$2$0.next = 46;
            break;
          }

          m$1$0 = __at___$1.reactor.render0(__at___$1.self, __at___$1.args, true);
          if (!getChecker(Promise)(m$1$0)) {
            context$2$0.next = 38;
            break;
          }

          p$0 = m$1$0;context$2$0.next = 35;
          return p$0;

        case 35:
          $targ$1 = context$2$0.sent;
          context$2$0.next = 40;
          break;

        case 38:
          x$2 = m$1$0;$targ$1 = x$2;

        case 40:
          t0$5 = $targ$1;__at___$1.replacement = t0$5;$get$0$0 = t0$5;return context$2$0.abrupt("return", $get$0$0);

        case 46:
          emit$0("beginCache", __at___$1.reactor.renderfn, __at___$1.self, __at___$1.args);m$2 = null;$2: for (_iterator = __at___$1.effects[Symbol.iterator](); !(_step = _iterator.next()).done;) {
            m$2 = _step.value;
            effect$0 = undefined;
            effect$0 = m$2;emit$0("effect", effect$0, false);
          }emit$0("endCache", __at___$1.reactor.renderfn, __at___$1.self, __at___$1.args);return context$2$0.abrupt("return", __at___$1);

        case 51:
        case "end":
          return context$2$0.stop();
      }
    }, callee$1$0, this);
  }), true);
};__amp____colon__(CachedResult$0, __amp____colon__(($targ$2 = "CachedResult", accum$0 = {}, accum$0["::name"] = $targ$2, accum$0), ($targ$3 = true, accum$1 = {}, accum$1["::egclass"] = $targ$3, accum$1)));frame__minus__stack$0 = [];current__minus__frame$0 = function currentFrame() {
  return send(frame__minus__stack$0, frame__minus__stack$0.length - 1);
};current__minus__context$0 = function currentContext() {
  var tmp$0 = undefined;tmp$0 = current__minus__frame$0();if (tmp$0 === undefined || tmp$0 === null) {
    return tmp$0;
  } else {
    return tmp$0.context;
  }
};side__minus__effect$0 = function sideEffect(effect$1) {
  var f$0 = undefined;var tmp$2 = undefined;var tmp$1 = undefined;if ((f$0 = current__minus__frame$0(), f$0)) {
    f$0.effects.push(effect$1);tmp$2 = send(f$0, "context", true);if (tmp$2 === undefined || tmp$2 === null) {
      tmp$1 = tmp$2;
    } else {
      tmp$1 = tmp$2.events;
    }if (tmp$1 === undefined || tmp$1 === null) {
      return tmp$1;
    } else {
      return tmp$1.emit("effect", effect$1, true);
    }
  }
};emit$0 = function emit() {
  var tmp$4 = undefined;var tmp$3 = undefined;var args$1 = undefined;var t0$6 = undefined;var m$3$0 = undefined;m$3$0 = arguments;t0$6 = m$3$0.length;if (t0$6 >= 0) {
    args$1 = Array.prototype.slice.call(m$3$0, 0);tmp$4 = current__minus__context$0();if (tmp$4 === undefined || tmp$4 === null) {
      tmp$3 = tmp$4;
    } else {
      tmp$3 = tmp$4.events;
    }if (tmp$3 === undefined || tmp$3 === null) {
      return tmp$3;
    } else {
      return send(send(tmp$3, "emit", true), args$1);
    }
  } else {
    return ___match_error(m$3$0, "{*args}");
  }
};with__minus__frame$0 = function withFrame(frame$0, fn$0) {
  return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
    var x$3, p$1, m$4$0, res$0;
    return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
      while (1) switch (context$2$0.prev = context$2$0.next) {
        case 0:
          x$3 = undefined;
          p$1 = undefined;
          m$4$0 = undefined;
          res$0 = undefined;
          frame__minus__stack$0.push(frame$0);m$4$0 = fn$0();
          if (!getChecker(Promise)(m$4$0)) {
            context$2$0.next = 13;
            break;
          }

          p$1 = m$4$0;context$2$0.next = 10;
          return p$1;

        case 10:
          res$0 = context$2$0.sent;
          context$2$0.next = 15;
          break;

        case 13:
          x$3 = m$4$0;res$0 = x$3;

        case 15:
          frame__minus__stack$0.pop();return context$2$0.abrupt("return", res$0);

        case 17:
        case "end":
          return context$2$0.stop();
      }
    }, callee$1$0, this);
  }), true);
};Frame$0 = function Frame() {
  var prev$0 = undefined;var $targ$4 = undefined;var context$0 = undefined;var t0$7 = undefined;var m$5$0 = undefined;var __at___$2 = undefined;if (!getChecker(Frame$0)(this)) {
    __at___$2 = Object.create(Frame$0.prototype);
  } else {
    __at___$2 = this;
  }m$5$0 = arguments;t0$7 = m$5$0.length;if (t0$7 >= 2 && t0$7 <= 3) {
    __at___$2.tracks = m$5$0[0];__at___$2.effects = m$5$0[1];if (2 >= t0$7) {
      context$0 = {};
    } else {
      context$0 = m$5$0[2];
    }prev$0 = current__minus__context$0() || {};$targ$4 = __amp__(__amp__(prev$0, { root: false }), context$0);__at___$2.context = $targ$4;void 0;
  } else {
    ___match_error(m$5$0, "{@tracks, @effects, context = {=}}");
  }return __at___$2;
};__amp____colon__(Frame$0, __amp____colon__(($targ$5 = "Frame", accum$2 = {}, accum$2["::name"] = $targ$5, accum$2), ($targ$6 = true, accum$3 = {}, accum$3["::egclass"] = $targ$6, accum$3)));_reprox$0 = function _reprox(temp$9$0, deps$0) {
  var t0$8 = undefined;var t0$9 = undefined;var bridge$$533$0 = undefined;var bridge$$532$0 = undefined;var t0$10 = undefined;var m$6 = undefined;var acc$0 = undefined;var temp$10 = undefined;var bridge$$538$0 = undefined;var bridge$$537$0 = undefined;var m$7 = undefined;var rval$0 = undefined;var bridge$$531$0 = undefined;var bridge$$536$0 = undefined;var x$4 = undefined;var ph$2$0 = undefined;t0$8 = temp$9$0;x$4 = t0$8;ph$2$0 = t0$8;bridge$$531$0 = ph$2$0;if ((bridge$$532$0 = bridge$$531$0, (bridge$$533$0 = bridge$$532$0, (t0$9 = ID$0, ___hasprop(bridge$$533$0, t0$9) && typeof send(bridge$$533$0, t0$9) === "number") || bridge$$533$0 === null) || bridge$$532$0 === void 0) || (t0$10 = symbols$0.notInspectable, ___hasprop(bridge$$531$0, t0$10) && (send(bridge$$531$0, t0$10) ? true : false))) {
    return reprox$0(x$4, deps$0, null);
  } else {
    if (Array.isArray(ph$2$0)) {
      acc$0 = [];temp$10 = null;m$6 = null;$3: for (var _iterator = x$4[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
        m$6 = _step.value;
        var y$0 = undefined;y$0 = m$6;temp$10 = _reprox$0(y$0, deps$0);acc$0.push(temp$10);
      }return acc$0;
    } else {
      bridge$$536$0 = ph$2$0;if ((bridge$$537$0 = bridge$$536$0, (bridge$$538$0 = bridge$$537$0, typeof bridge$$538$0 === "number" || typeof bridge$$538$0 === "string") || typeof bridge$$537$0 === "boolean") || getChecker(Symbol)(bridge$$536$0)) {
        return x$4;
      } else {
        rval$0 = Object.create(Object.getPrototypeOf(x$4));m$7 = null;$4: for (var _iterator2 = items(x$4)[Symbol.iterator](), _step2; !(_step2 = _iterator2.next()).done;) {
          m$7 = _step2.value;
          var $targ$7 = undefined;var k$0 = undefined;var v$0 = undefined;var t0$11 = undefined;var t1$2 = undefined;t0$11 = m$7;if (Array.isArray(t0$11) && (t1$2 = t0$11.length, t1$2 === 2)) {
            k$0 = t0$11[0];v$0 = t0$11[1];$targ$7 = _reprox$0(v$0, deps$0);rval$0[k$0] = $targ$7;void 0;
          } else {
            ___match_error(m$7);
          }
        }return rval$0;
      }
    }
  }
};Reactor$0 = function Reactor(temp$11$0) {
  var $targ$8 = undefined;var __at___$3 = undefined;if (!getChecker(Reactor$0)(this)) {
    __at___$3 = Object.create(Reactor$0.prototype);
  } else {
    __at___$3 = this;
  }__at___$3.renderfn = temp$11$0;$targ$8 = {};__at___$3.argmap = $targ$8;return __at___$3;
};Reactor$0.prototype.render0 = function render0() {
  var the__minus__self$0 = undefined;var args$2 = undefined;var force$0 = undefined;var t0$12 = undefined;var m$8$0 = undefined;var __at___$4 = undefined;var self$1 = undefined;__at___$4 = this;self$1 = this;m$8$0 = arguments;t0$12 = m$8$0.length;if (t0$12 >= 2 && t0$12 <= 3) {
    the__minus__self$0 = m$8$0[0];args$2 = m$8$0[1];if (2 >= t0$12) {
      force$0 = false;
    } else {
      force$0 = m$8$0[2];
    }return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
      var x$5, p$2, m$10$0, end__minus__time$0, rval$1, start__minus__time$0, res$1, x$6, p$3, m$11$0, c$0, cached$2, bridge$$731$0, m$9$0, key$0, cached$1, $targ$9, $targ$10, e$0;
      return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
        while (1) switch (context$2$0.prev = context$2$0.next) {
          case 0:
            x$5 = undefined;
            p$2 = undefined;
            m$10$0 = undefined;
            end__minus__time$0 = undefined;
            rval$1 = undefined;
            start__minus__time$0 = undefined;
            res$1 = undefined;
            x$6 = undefined;
            p$3 = undefined;
            m$11$0 = undefined;
            c$0 = undefined;
            cached$2 = undefined;
            bridge$$731$0 = undefined;
            m$9$0 = undefined;
            key$0 = undefined;
            cached$1 = undefined;
            $targ$9 = undefined;
            key$0 = args__minus__key$0([the__minus__self$0].concat(args$2));m$9$0 = send(send(__at___$4, "argmap", true), key$0);bridge$$731$0 = m$9$0;
            if (!(bridge$$731$0 === void 0 || force$0)) {
              context$2$0.next = 55;
              break;
            }

            delete __at___$4.argmap[key$0];emit$0("startCompute", __at___$4.renderfn, the__minus__self$0, args$2);start__minus__time$0 = new Date().valueOf();rval$1 = false;context$2$0.prev = 25;
            m$10$0 = __at___$4.renderCacheObject(the__minus__self$0, args$2);
            if (!getChecker(Promise)(m$10$0)) {
              context$2$0.next = 34;
              break;
            }

            p$2 = m$10$0;context$2$0.next = 31;
            return p$2;

          case 31:
            rval$1 = context$2$0.sent;
            context$2$0.next = 36;
            break;

          case 34:
            x$5 = m$10$0;rval$1 = x$5;

          case 36:
            rval$1;context$2$0.next = 47;
            break;

          case 39:
            context$2$0.prev = 39;
            context$2$0.t0 = context$2$0["catch"](25);
            $targ$10 = undefined;
            e$0 = undefined;
            e$0 = context$2$0.t0;if (!e$0._quasarLogged) {
              emit$0("exception", __at___$4.renderfn, the__minus__self$0, args$2, e$0);$targ$10 = true;e$0._quasarLogged = $targ$10;void 0;
            }throw e$0;

          case 47:
            context$2$0.prev = 47;
            end__minus__time$0 = new Date().valueOf();emit$0("endCompute", __at___$4.renderfn, the__minus__self$0, args$2, end__minus__time$0 - start__minus__time$0);return context$2$0.finish(47);

          case 51:
            res$1 = rval$1;cached$1 = res$1;context$2$0.next = 67;
            break;

          case 55:
            cached$2 = m$9$0;c$0 = get__minus__replacement$0(cached$2);m$11$0 = c$0.update();
            if (!getChecker(Promise)(m$11$0)) {
              context$2$0.next = 65;
              break;
            }

            p$3 = m$11$0;context$2$0.next = 62;
            return p$3;

          case 62:
            cached$1 = context$2$0.sent;
            context$2$0.next = 67;
            break;

          case 65:
            x$6 = m$11$0;cached$1 = x$6;

          case 67:
            $targ$9 = cached$1;__at___$4.argmap[key$0] = $targ$9;return context$2$0.abrupt("return", cached$1);

          case 70:
          case "end":
            return context$2$0.stop();
        }
      }, callee$1$0, this, [[25, 39, 47, 51]]);
    }), true);
  } else {
    return ___match_error(m$8$0, "{the-self, args, force = false}");
  }
};Reactor$0.prototype.render = function render() {
  var the__minus__self$1 = undefined;var args$3 = undefined;var force$1 = undefined;var t0$13 = undefined;var m$12$0 = undefined;var __at___$5 = undefined;var self$2 = undefined;__at___$5 = this;self$2 = this;m$12$0 = arguments;t0$13 = m$12$0.length;if (t0$13 >= 2 && t0$13 <= 3) {
    the__minus__self$1 = m$12$0[0];args$3 = m$12$0[1];if (2 >= t0$13) {
      force$1 = false;
    } else {
      force$1 = m$12$0[2];
    }return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
      var x$7, p$4, m$13$0, tmp$5, m$14, cached$3, parent$0, _iterator, _step, tmp$6, x$8;

      return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
        while (1) switch (context$2$0.prev = context$2$0.next) {
          case 0:
            x$7 = undefined;
            p$4 = undefined;
            m$13$0 = undefined;
            tmp$5 = undefined;
            m$14 = undefined;
            cached$3 = undefined;
            parent$0 = undefined;
            m$13$0 = __at___$5.render0(the__minus__self$1, args$3, force$1);
            if (!getChecker(Promise)(m$13$0)) {
              context$2$0.next = 15;
              break;
            }

            p$4 = m$13$0;context$2$0.next = 12;
            return p$4;

          case 12:
            cached$3 = context$2$0.sent;
            context$2$0.next = 17;
            break;

          case 15:
            x$7 = m$13$0;cached$3 = x$7;

          case 17:
            parent$0 = current__minus__frame$0();tmp$5 = parent$0;if (tmp$5 === undefined || tmp$5 === null) {
              tmp$5;
            } else {
              tmp$5.tracks.merge(cached$3.tracks);
            }m$14 = null;$5: for (_iterator = cached$3.effects[Symbol.iterator](); !(_step = _iterator.next()).done;) {
              m$14 = _step.value;
              tmp$6 = undefined;
              x$8 = undefined;
              x$8 = m$14;tmp$6 = parent$0;if (tmp$6 === undefined || tmp$6 === null) {
                tmp$6;
              } else {
                tmp$6.effects.push(x$8);
              }
            }return context$2$0.abrupt("return", cached$3);

          case 23:
          case "end":
            return context$2$0.stop();
        }
      }, callee$1$0, this);
    }), true);
  } else {
    return ___match_error(m$12$0, "{the-self, args, force = false}");
  }
};Reactor$0.prototype.renderCacheObject = function (the__minus__self$2, args$4) {
  var __at___$6 = undefined;var self$3 = undefined;__at___$6 = this;self$3 = this;return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
    var m$15, acc$1, temp$12, t0$14, t1$3, x$10, p$6, m$16$0, f$1, tracks$0, effects$0, pargs$0, pself$0, $targ$11, version$1, res$2, cached$4, _iterator, _step, arg$0;

    return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
      while (1) switch (context$2$0.prev = context$2$0.next) {
        case 0:
          m$15 = undefined;
          acc$1 = undefined;
          temp$12 = undefined;
          t0$14 = undefined;
          t1$3 = undefined;
          x$10 = undefined;
          p$6 = undefined;
          m$16$0 = undefined;
          f$1 = undefined;
          tracks$0 = undefined;
          effects$0 = undefined;
          pargs$0 = undefined;
          pself$0 = undefined;
          $targ$11 = undefined;
          version$1 = undefined;
          res$2 = undefined;
          cached$4 = undefined;
          tracks$0 = Tracks$0();effects$0 = [];acc$1 = [];temp$12 = null;m$15 = null;$6: for (_iterator = args$4[Symbol.iterator](); !(_step = _iterator.next()).done;) {
            m$15 = _step.value;
            arg$0 = undefined;
            arg$0 = m$15;temp$12 = _reprox$0(arg$0, tracks$0);acc$1.push(temp$12);
          }pargs$0 = acc$1;pself$0 = reprox$0(the__minus__self$2, tracks$0, null);$targ$11 = current__minus__context$0();t0$14 = $targ$11;if (___hasprop(t0$14, "dir")) {
            t1$3 = t0$14.dir;if (___hasprop(t1$3, "version")) {
              version$1 = t1$3.version;
            } else {
              version$1 = null;
            }
          } else {
            ___match_error($targ$11, "{\n         dir => {=> version = null}\n         ;; => out-dir\n      }");
          }f$1 = function f() {
            return spawn.call(this, regeneratorRuntime.mark(function callee$3$0() {
              var x$9, p$5, m$17$0;
              return regeneratorRuntime.wrap(function callee$3$0$(context$4$0) {
                while (1) switch (context$4$0.prev = context$4$0.next) {
                  case 0:
                    x$9 = undefined;
                    p$5 = undefined;
                    m$17$0 = undefined;
                    m$17$0 = __at___$6.renderfn.apply(pself$0, pargs$0);
                    if (!getChecker(Promise)(m$17$0)) {
                      context$4$0.next = 11;
                      break;
                    }

                    p$5 = m$17$0;context$4$0.next = 8;
                    return p$5;

                  case 8:
                    return context$4$0.abrupt("return", context$4$0.sent);

                  case 11:
                    x$9 = m$17$0;return context$4$0.abrupt("return", x$9);

                  case 13:
                  case "end":
                    return context$4$0.stop();
                }
              }, callee$3$0, this);
            }), true);
          };m$16$0 = with__minus__frame$0(Frame$0(tracks$0, effects$0), f$1);
          if (!getChecker(Promise)(m$16$0)) {
            context$2$0.next = 37;
            break;
          }

          p$6 = m$16$0;context$2$0.next = 34;
          return p$6;

        case 34:
          res$2 = context$2$0.sent;
          context$2$0.next = 39;
          break;

        case 37:
          x$10 = m$16$0;res$2 = x$10;

        case 39:
          cached$4 = CachedResult$0(__at___$6, the__minus__self$2, args$4, res$2, tracks$0, effects$0, version$1);return context$2$0.abrupt("return", cached$4);

        case 41:
        case "end":
          return context$2$0.stop();
      }
    }, callee$1$0, this);
  }), true);
};__amp____colon__(Reactor$0, __amp____colon__(($targ$12 = "Reactor", accum$4 = {}, accum$4["::name"] = $targ$12, accum$4), ($targ$13 = true, accum$5 = {}, accum$5["::egclass"] = $targ$13, accum$5)));cached__minus__function$0 = function cachedFunction(render$0) {
  var r$0 = undefined;var mkcache$0 = undefined;var rval$2 = undefined;var $targ$14 = undefined;var $targ$15 = undefined;var $targ$16 = undefined;r$0 = Reactor$0(render$0);mkcache$0 = function mkcache(args$5) {
    var t0$15 = undefined;var $targ$17 = undefined;var __at___$7 = undefined;var self$4 = undefined;$targ$17 = this;t0$15 = $targ$17;__at___$7 = t0$15;self$4 = t0$15;[__at___$7, self$4];return r$0.render(__at___$7 || current__minus__context$0().state.self(), args$5);
  };rval$2 = function rval() {
    var args$6 = undefined;var t0$16 = undefined;var m$18$0 = undefined;m$18$0 = arguments;t0$16 = m$18$0.length;if (t0$16 >= 0) {
      args$6 = Array.prototype.slice.call(m$18$0, 0);return spawn.call(this, regeneratorRuntime.mark(function callee$2$0() {
        var x$11, p$7, m$21$0;
        return regeneratorRuntime.wrap(function callee$2$0$(context$3$0) {
          while (1) switch (context$3$0.prev = context$3$0.next) {
            case 0:
              x$11 = undefined;
              p$7 = undefined;
              m$21$0 = undefined;
              m$21$0 = mkcache$0(args$6);

              if (!getChecker(Promise)(m$21$0)) {
                context$3$0.next = 11;
                break;
              }

              p$7 = m$21$0;
              context$3$0.next = 8;
              return p$7;

            case 8:
              context$3$0.t1 = context$3$0.sent;
              context$3$0.next = 12;
              break;

            case 11:
              context$3$0.t1 = (x$11 = m$21$0, x$11);

            case 12:
              return context$3$0.abrupt("return", context$3$0.t1.value);

            case 13:
            case "end":
              return context$3$0.stop();
          }
        }, callee$2$0, this);
      }), true);
    } else {
      return ___match_error(m$18$0, "{*args}");
    }
  };$targ$14 = render$0;rval$2.orig = $targ$14;$targ$15 = r$0;rval$2.reactor = $targ$15;$targ$16 = function () {
    var args$7 = undefined;var t0$17 = undefined;var m$22$0 = undefined;m$22$0 = arguments;t0$17 = m$22$0.length;if (t0$17 >= 0) {
      args$7 = Array.prototype.slice.call(m$22$0, 0);return mkcache$0(args$7);
    } else {
      return ___match_error(m$22$0, "{*args}");
    }
  };rval$2.rend = $targ$16;return rval$2;
};$targ$18 = arg__minus__key$0;exports.argKey = $targ$18;$targ$19 = args__minus__key$0;exports.argsKey = $targ$19;$targ$20 = cached__minus__function$0;exports.cachedFunction = $targ$20;$targ$21 = Frame$0;exports.Frame = $targ$21;$targ$22 = with__minus__frame$0;exports.withFrame = $targ$22;$targ$23 = current__minus__frame$0;exports.currentFrame = $targ$23;$targ$24 = current__minus__context$0;exports.currentContext = $targ$24;$targ$25 = side__minus__effect$0;exports.sideEffect = $targ$25;$targ$26 = emit$0;exports.emit = $targ$26;void 0;
