// quasar/lib/proxy.js
"use strict";

require("earlgrey-runtime/5");var $targ$15 = undefined;var accum$3 = undefined;var $targ$16 = undefined;var accum$4 = undefined;var $targ$17 = undefined;var accum$5 = undefined;var $targ$18 = undefined;var $targ$19 = undefined;var $targ$20 = undefined;var $targ$21 = undefined;var $targ$22 = undefined;var $targ$23 = undefined;var $targ$24 = undefined;var $targ$25 = undefined;var util$0 = undefined;var tuck$0 = undefined;var identify$0 = undefined;var next__minus__id$0 = undefined;var ID$0 = undefined;var FORKID$0 = undefined;var PROTOID$0 = undefined;var VERSION$0 = undefined;var DIRECTORY$0 = undefined;var symbols$0 = undefined;var $0$0 = undefined;var Tracks$0 = undefined;var Patch$0 = undefined;var ObjectTrack$0 = undefined;var ArrayTrack$0 = undefined;var kaiser$0 = undefined;var get__minus__handler$0 = undefined;var extend__minus__handler$0 = undefined;var proxy$0 = undefined;var deprox$0 = undefined;var reprox$0 = undefined;var read__minus__proxy$0 = undefined;var write__minus__proxy$0 = undefined;var setup__minus__proxy$0 = undefined;var basic__minus__object__minus__proxy$0 = undefined;var basic__minus__get$0 = undefined;var basic__minus__set$0 = undefined;var proxy__minus__proto__minus__cache$0 = undefined;var proxy__minus__proto__minus__for$0 = undefined;var object__minus__proxy$0 = undefined;var array__minus__proxy$0 = undefined;var $targ$0 = undefined;var $targ$1 = undefined;var $targ$2 = undefined;util$0 = require("./util");tuck$0 = getProperty(util$0, "tuck", "util");identify$0 = getProperty(util$0, "identify", "util");next__minus__id$0 = getProperty(util$0, "nextId", "util");ID$0 = getProperty(util$0, "ID", "util");FORKID$0 = getProperty(util$0, "FORKID", "util");PROTOID$0 = getProperty(util$0, "PROTOID", "util");VERSION$0 = getProperty(util$0, "VERSION", "util");DIRECTORY$0 = getProperty(util$0, "DIRECTORY", "util");symbols$0 = getProperty(util$0, "symbols", "util");$0$0 = require("./tracks");Tracks$0 = getProperty($0$0, "Tracks", "./tracks");Patch$0 = getProperty($0$0, "Patch", "./tracks");ObjectTrack$0 = getProperty($0$0, "ObjectTrack", "./tracks");ArrayTrack$0 = getProperty($0$0, "ArrayTrack", "./tracks");kaiser$0 = require("kaiser");get__minus__handler$0 = getProperty(kaiser$0, "getHandler", "kaiser");extend__minus__handler$0 = getProperty(kaiser$0, "extendHandler", "kaiser");proxy$0 = function proxy(temp$0$0, tracks$0, patch$0) {
  var t0$0 = undefined;var t1$0 = undefined;var t2$0 = undefined;var fn$0 = undefined;var t0$1 = undefined;var value$0 = undefined;var h$0 = undefined;var ph$0$0 = undefined;t0$0 = temp$0$0;value$0 = t0$0;t1$0 = getProjector(get__minus__handler$0)(t0$0);if (t1$0[0]) {
    t2$0 = t1$0[1];h$0 = t2$0;ph$0$0 = t2$0;
  } else {
    ___match_error(temp$0$0);
  }t0$1 = ph$0$0;if (___hasprop(t0$1, "proxy")) {
    fn$0 = t0$1.proxy;return fn$0(value$0, tracks$0, patch$0);
  } else {
    throw ErrorFactory(["quasar", "proxy"]).create("Cannot proxy " + value$0);
  }
};deprox$0 = function deprox(x$0) {
  if (x$0 && send(x$0, symbols$0.source)) {
    return send(x$0, symbols$0.source);
  } else {
    return x$0;
  }
};reprox$0 = function reprox(temp$1$0, tracks$1, patch$1) {
  var t0$2 = undefined;var bridge$$4648$0 = undefined;var bridge$$4647$0 = undefined;var bridge$$4646$0 = undefined;var bridge$$4645$0 = undefined;var obj$0 = undefined;var bridge$$4644$0 = undefined;var t0$3 = undefined;var t1$1 = undefined;var t2$1 = undefined;var value$1 = undefined;var ph$1$0 = undefined;t0$2 = temp$1$0;value$1 = t0$2;ph$1$0 = t0$2;bridge$$4644$0 = ph$1$0;if ((bridge$$4645$0 = bridge$$4644$0, (bridge$$4646$0 = bridge$$4645$0, (bridge$$4647$0 = bridge$$4646$0, (bridge$$4648$0 = bridge$$4647$0, bridge$$4648$0 === null || bridge$$4648$0 === void 0) || typeof bridge$$4647$0 === "number") || typeof bridge$$4646$0 === "string") || getChecker(Symbol)(bridge$$4645$0)) || typeof bridge$$4644$0 === "boolean") {
    return value$1;
  } else {
    t0$3 = ph$1$0;t1$1 = symbols$0.source;if (___hasprop(t0$3, t1$1) && (t2$1 = send(t0$3, t1$1), t2$1 ? true : false)) {
      obj$0 = t2$1;return proxy$0(obj$0, tracks$1 || send(value$1, symbols$0.tracks), patch$1 || send(value$1, symbols$0.patch));
    } else {
      return proxy$0(value$1, tracks$1, patch$1);
    }
  }
};read__minus__proxy$0 = function readProxy(value$2, tracks$2) {
  return proxy$0(value$2, tracks$2, null);
};write__minus__proxy$0 = function writeProxy(value$3, patch$2) {
  return proxy$0(value$3, null, patch$2);
};setup__minus__proxy$0 = function setupProxy(self$0, obj$1, tracks$3, patch$3, track__minus__initializer$0) {
  var $targ$3 = undefined;var $targ$4 = undefined;var $targ$5 = undefined;var id$0 = undefined;var $targ$6 = undefined;var $targ$7 = undefined;var $targ$8 = undefined;track__minus__initializer$0 = track__minus__initializer$0 || ObjectTrack$0;identify$0(obj$1);$targ$3 = obj$1;self$0[symbols$0.obj] = $targ$3;$targ$4 = tracks$3;self$0[symbols$0.tracks] = $targ$4;$targ$5 = patch$3;self$0[symbols$0.patch] = $targ$5;id$0 = send(obj$1, ID$0);$targ$6 = function (key$0) {
    var tracks$4 = undefined;tracks$4 = send(self$0, symbols$0.tracks);if (tracks$4) {
      return tracks$4.mark(self$0, key$0, track__minus__initializer$0);
    }
  };self$0[symbols$0.registerTracks] = $targ$6;$targ$7 = function (key$1, val$0) {
    var patch$4 = undefined;patch$4 = send(self$0, symbols$0.patch);if (patch$4) {
      return patch$4.register(self$0, key$1, val$0, function () {
        return {};
      });
    }
  };self$0[symbols$0.registerPatch] = $targ$7;$targ$8 = function (key$2, fn$1) {
    var tmp$0 = undefined;var d$0 = undefined;var patch$5 = undefined;patch$5 = send(self$0, symbols$0.patch);if (patch$5) {
      tmp$0 = patch$5.getFor(obj$1);if (tmp$0 === undefined || tmp$0 === null) {
        d$0 = tmp$0;
      } else {
        d$0 = send(tmp$0, key$2);
      }return patch$5.register(self$0, key$2, fn$1(d$0), function () {
        return {};
      });
    }
  };self$0[symbols$0.registerPatchFunction] = $targ$8;tuck$0(self$0, ID$0, send(obj$1, ID$0));tuck$0(self$0, symbols$0.source, obj$1);return tuck$0(self$0, VERSION$0, send(obj$1, VERSION$0));
};basic__minus__object__minus__proxy$0 = function basicObjectProxy(obj$2, tracks$5, patch$6) {
  var self$1 = undefined;self$1 = this;setup__minus__proxy$0(self$1, obj$2, tracks$5, patch$6, ObjectTrack$0);return Object.preventExtensions(self$1);
};basic__minus__get$0 = function basicGet(name$0) {
  var self$2 = undefined;self$2 = this;send(self$2, symbols$0.registerTracks, true)(name$0);return proxy$0(send(send(self$2, symbols$0.obj, true), name$0), send(self$2, symbols$0.tracks), send(self$2, symbols$0.patch));
};basic__minus__set$0 = function basicSet(name$1, new__minus__value$0) {
  var self$3 = undefined;self$3 = this;return send(self$3, symbols$0.registerPatch, true)(name$1, ["update", deprox$0(new__minus__value$0)]);
};proxy__minus__proto__minus__cache$0 = {};proxy__minus__proto__minus__for$0 = function proxyProtoFor(obj$3) {
  var pid$1 = undefined;var $$4858$0 = undefined;var bridge$$4855$0 = undefined;var m$1$0 = undefined;var m$2 = undefined;var proto$0 = undefined;var $targ$9 = undefined;var $targ$10 = undefined;var $targ$11 = undefined;var proto$1 = undefined;var m$0$0 = undefined;var orig__minus__proto$0 = undefined;var pid$0 = undefined;var the__minus__keys$0 = undefined;var key$3 = undefined;orig__minus__proto$0 = Object.getPrototypeOf(obj$3);m$1$0 = orig__minus__proto$0;bridge$$4855$0 = m$1$0;if (bridge$$4855$0 === null || bridge$$4855$0 === void 0) {
    pid$0 = null;
  } else {
    if (Object.hasOwnProperty.call(orig__minus__proto$0, PROTOID$0)) {
      pid$0 = send(orig__minus__proto$0, PROTOID$0);
    } else {
      pid$1 = next__minus__id$0();tuck$0(orig__minus__proto$0, PROTOID$0, pid$1);pid$0 = pid$1;
    }
  }the__minus__keys$0 = keys(obj$3).sort();key$3 = "" + pid$0 + "\u0001" + the__minus__keys$0.join("\u0000");m$0$0 = send(proxy__minus__proto__minus__cache$0, key$3);if (m$0$0 === void 0) {
    proto$0 = Object.create(orig__minus__proto$0);m$2 = null;$1: for (var _iterator = the__minus__keys$0[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      (function () {
        m$2 = _step.value;
        var accum$0 = undefined;var accum$1 = undefined;var accum$2 = undefined;var name$2 = undefined;var ph$2$0 = undefined;var t0$4 = undefined;t0$4 = m$2;name$2 = t0$4;ph$2$0 = t0$4;if (getChecker(RegExp("^__", ""))(ph$2$0)) {
          Object.defineProperty(proto$0, name$2, __amp____colon__((accum$0 = {}, accum$0.get = function get() {
            return send(send(this, symbols$0.obj, true), name$2);
          }, accum$0), { enumerable: true }));
        } else {
          Object.defineProperty(proto$0, name$2, __amp____colon__((accum$1 = {}, accum$1.get = function get() {
            return this._get(name$2);
          }, accum$1), __amp____colon__((accum$2 = {}, accum$2.set = function set(value$4) {
            return this._set(name$2, value$4);
          }, accum$2), { enumerable: true })));
        }
      })();
    }$targ$9 = basic__minus__get$0;proto$0._get = $targ$9;$targ$10 = basic__minus__set$0;proto$0._set = $targ$10;$targ$11 = proto$0;proxy__minus__proto__minus__cache$0[key$3] = $targ$11;return proto$0;
  } else {
    proto$1 = m$0$0;return proto$1;
  }
};object__minus__proxy$0 = function objectProxy(obj$4, tracks$6, patch$7) {
  var p$0 = undefined;p$0 = proxy__minus__proto__minus__for$0(obj$4);return basic__minus__object__minus__proxy$0.call(Object.create(p$0), obj$4, tracks$6, patch$7);
};array__minus__proxy$0 = function arrayProxy(obj$5, tracks$7, patch$8) {
  var __at___$0 = undefined;if (!getChecker(array__minus__proxy$0)(this)) {
    __at___$0 = Object.create(array__minus__proxy$0.prototype);
  } else {
    __at___$0 = this;
  }setup__minus__proxy$0(__at___$0, obj$5, tracks$7, patch$8, ArrayTrack$0);return __at___$0;
};array__minus__proxy$0.prototype._updateIter = function () {
  var i2$0 = undefined;var j2$0 = undefined;var t0$6 = undefined;var t1$2 = undefined;var ph$4$0 = undefined;var i$0 = undefined;var j$0 = undefined;var stick__minus__to__minus__end$0 = undefined;var t0$5 = undefined;var m$3$0 = undefined;var __at___$1 = undefined;var self$4 = undefined;__at___$1 = this;self$4 = this;m$3$0 = arguments;t0$5 = m$3$0.length;if (t0$5 >= 3 && t0$5 <= 4) {
    ph$4$0 = m$3$0[0];i$0 = m$3$0[1];j$0 = m$3$0[2];if (3 >= t0$5) {
      stick__minus__to__minus__end$0 = false;
    } else {
      stick__minus__to__minus__end$0 = m$3$0[3];
    }if (ph$4$0 === void 0) {
      return [i$0, j$0, stick__minus__to__minus__end$0];
    } else {
      t0$6 = ph$4$0;t1$2 = t0$6.length;if (t1$2 === 2) {
        i2$0 = t0$6[0];j2$0 = t0$6[1];return [Math.min(i$0, i2$0), Math.max(j$0, j2$0), stick__minus__to__minus__end$0];
      } else {
        return [i$0, j$0, stick__minus__to__minus__end$0];
      }
    }
  } else {
    return ___match_error(m$3$0, "{match, i, j, stick-to-end = false}");
  }
};array__minus__proxy$0.prototype.get = function get(i$1) {
  var __at___$2 = undefined;var self$5 = undefined;__at___$2 = this;self$5 = this;send(self$5, symbols$0.registerTracks, true)(i$1);return proxy$0(send(send(self$5, symbols$0.obj, true), i$1), send(self$5, symbols$0.tracks), send(self$5, symbols$0.patch));
};array__minus__proxy$0.prototype.slice = function slice() {
  var stick$0 = undefined;var i$2 = undefined;var j$1 = undefined;var t0$7 = undefined;var m$4$0 = undefined;var __at___$3 = undefined;var self$6 = undefined;__at___$3 = this;self$6 = this;m$4$0 = arguments;t0$7 = m$4$0.length;if (t0$7 >= 0 && t0$7 <= 2) {
    if (0 >= t0$7) {
      i$2 = 0;
    } else {
      i$2 = m$4$0[0];
    }if (1 >= t0$7) {
      j$1 = null;
    } else {
      j$1 = m$4$0[1];
    }if (j$1 === null) {
      j$1 = send(self$6, symbols$0.obj).length;stick$0 = true;
    } else {
      stick$0 = false;
    }send(self$6, symbols$0.registerTracks, true)("@iter");return regeneratorRuntime.mark(function callee$1$0() {
      var m$5, acc$0, temp$2, _iterator, _step, x$1;

      return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
        while (1) switch (context$2$0.prev = context$2$0.next) {
          case 0:
            m$5 = undefined;
            acc$0 = undefined;
            temp$2 = undefined;
            acc$0 = [];temp$2 = null;m$5 = null;_iterator = send(self$6, symbols$0.obj).slice(i$2, j$1)[Symbol.iterator]();

          case 7:
            if ((_step = _iterator.next()).done) {
              context$2$0.next = 16;
              break;
            }

            m$5 = _step.value;
            x$1 = undefined;
            x$1 = m$5;temp$2 = proxy$0(x$1, send(self$6, symbols$0.tracks), send(self$6, symbols$0.patch));context$2$0.next = 14;
            return temp$2;

          case 14:
            context$2$0.next = 7;
            break;

          case 16:
            return context$2$0.abrupt("return", acc$0);

          case 17:
          case "end":
            return context$2$0.stop();
        }
      }, callee$1$0, this);
    })();
  } else {
    return ___match_error(m$4$0, "{i = 0, var j = null}");
  }
};$targ$0 = function () {
  var m$6 = undefined;var acc$1 = undefined;var temp$3 = undefined;var proxies$0 = undefined;var __at___$4 = undefined;var self$7 = undefined;__at___$4 = this;self$7 = this;acc$1 = [];temp$3 = null;m$6 = null;$3: for (var _iterator = send(self$7, symbols$0.obj)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$6 = _step.value;
    var x$2 = undefined;x$2 = m$6;temp$3 = proxy$0(x$2, send(self$7, symbols$0.tracks), send(self$7, symbols$0.patch));acc$1.push(temp$3);
  }proxies$0 = acc$1;send(self$7, symbols$0.registerTracks, true)("@iter");return send(proxies$0, Symbol.iterator, true)();
};array__minus__proxy$0.prototype[Symbol.iterator] = $targ$0;array__minus__proxy$0.prototype.forEach = function (fn$2) {
  var m$7 = undefined;var __at___$5 = undefined;var self$8 = undefined;__at___$5 = this;self$8 = this;m$7 = null;$4: for (var _iterator = __at___$5[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$7 = _step.value;
    var x$3 = undefined;x$3 = m$7;fn$2(x$3);
  }return undefined;
};array__minus__proxy$0.prototype.indexOf = function (temp$4$0) {
  var t0$8 = undefined;var elem$0 = undefined;var __at___$6 = undefined;var self$9 = undefined;__at___$6 = this;self$9 = this;t0$8 = getProjector(deprox$0)(temp$4$0);if (t0$8[0]) {
    elem$0 = t0$8[1];
  } else {
    ___match_error(temp$4$0);
  }send(self$9, symbols$0.registerTracks, true)("@iter");return send(self$9, symbols$0.obj).indexOf(elem$0);
};array__minus__proxy$0.prototype.map = function map(fn$3) {
  var m$8 = undefined;var acc$2 = undefined;var temp$5 = undefined;var __at___$7 = undefined;var self$10 = undefined;__at___$7 = this;self$10 = this;acc$2 = [];temp$5 = null;m$8 = null;$5: for (var _iterator = __at___$7[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$8 = _step.value;
    var x$4 = undefined;x$4 = m$8;temp$5 = fn$3(x$4);acc$2.push(temp$5);
  }return acc$2;
};array__minus__proxy$0.prototype.join = function join(sep$0) {
  var m$9 = undefined;var acc$3 = undefined;var temp$6 = undefined;var __at___$8 = undefined;var self$11 = undefined;__at___$8 = this;self$11 = this;return (acc$3 = [], temp$6 = null, m$9 = null, (function () {
    $10: for (var _iterator = __at___$8[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      m$9 = _step.value;
      var x$5 = undefined;x$5 = m$9;temp$6 = x$5;acc$3.push(temp$6);
    }
  })(), acc$3).join(sep$0);
};array__minus__proxy$0.prototype.toString = function () {
  var __at___$9 = undefined;var self$12 = undefined;__at___$9 = this;self$12 = this;return "{" + __at___$9.join(", ") + "}";
};array__minus__proxy$0.prototype.set = function set(i$3, new__minus__value$1) {
  var __at___$10 = undefined;var self$13 = undefined;__at___$10 = this;self$13 = this;return __at___$10._splice(i$3, 1, [new__minus__value$1], false);
};array__minus__proxy$0.prototype.remove = function remove(temp$7$0) {
  var t0$9 = undefined;var idx$0 = undefined;var elem$1 = undefined;var __at___$11 = undefined;var self$14 = undefined;__at___$11 = this;self$14 = this;t0$9 = getProjector(deprox$0)(temp$7$0);if (t0$9[0]) {
    elem$1 = t0$9[1];
  } else {
    ___match_error(temp$7$0);
  }idx$0 = send(self$14, symbols$0.obj).indexOf(elem$1);if (idx$0 !== -1) {
    return __at___$11.splice(idx$0, 1);
  }
};array__minus__proxy$0.prototype.pop = function pop() {
  var n$0 = undefined;var t0$10 = undefined;var m$10$0 = undefined;var __at___$12 = undefined;var self$15 = undefined;__at___$12 = this;self$15 = this;m$10$0 = arguments;t0$10 = m$10$0.length;if (t0$10 >= 0 && t0$10 <= 1) {
    if (0 >= t0$10) {
      n$0 = 1;
    } else {
      n$0 = m$10$0[0];
    }return __at___$12._splice(send(self$15, symbols$0.obj).length - n$0, n$0, [], false);
  } else {
    return ___match_error(m$10$0, "{n = 1}");
  }
};array__minus__proxy$0.prototype.push = function push() {
  var values$0 = undefined;var t0$11 = undefined;var m$11$0 = undefined;var __at___$13 = undefined;var self$16 = undefined;__at___$13 = this;self$16 = this;m$11$0 = arguments;t0$11 = m$11$0.length;if (t0$11 >= 0) {
    values$0 = Array.prototype.slice.call(m$11$0, 0);return __at___$13._splice(send(self$16, symbols$0.obj).length, 0, values$0, "append");
  } else {
    return ___match_error(m$11$0, "{*values}");
  }
};array__minus__proxy$0.prototype.shift = function shift() {
  var n$1 = undefined;var t0$12 = undefined;var m$12$0 = undefined;var __at___$14 = undefined;var self$17 = undefined;__at___$14 = this;self$17 = this;m$12$0 = arguments;t0$12 = m$12$0.length;if (t0$12 >= 0 && t0$12 <= 1) {
    if (0 >= t0$12) {
      n$1 = 1;
    } else {
      n$1 = m$12$0[0];
    }return __at___$14._splice(0, n$1, [], false);
  } else {
    return ___match_error(m$12$0, "{n = 1}");
  }
};array__minus__proxy$0.prototype.unshift = function unshift() {
  var values$1 = undefined;var t0$13 = undefined;var m$13$0 = undefined;var __at___$15 = undefined;var self$18 = undefined;__at___$15 = this;self$18 = this;m$13$0 = arguments;t0$13 = m$13$0.length;if (t0$13 >= 0) {
    values$1 = Array.prototype.slice.call(m$13$0, 0);return __at___$15._splice(0, 0, values$1, "prepend");
  } else {
    return ___match_error(m$13$0, "{*values}");
  }
};$targ$1 = function () {
  var i$4 = undefined;var n$2 = undefined;var values$2 = undefined;var t0$14 = undefined;var m$14$0 = undefined;var __at___$16 = undefined;var self$19 = undefined;__at___$16 = this;self$19 = this;m$14$0 = arguments;t0$14 = m$14$0.length;if (t0$14 >= 2) {
    i$4 = m$14$0[0];n$2 = m$14$0[1];values$2 = Array.prototype.slice.call(m$14$0, 2);return __at___$16._splice(i$4, n$2, values$2, false);
  } else {
    return ___match_error(m$14$0, "{i, n, *values}");
  }
};array__minus__proxy$0.prototype.splice = $targ$1;array__minus__proxy$0.prototype.replace = function replace(new__minus__array$0) {
  var __at___$17 = undefined;var self$20 = undefined;__at___$17 = this;self$20 = this;return __at___$17._splice(0, send(self$20, symbols$0.obj).length, new__minus__array$0, false);
};$targ$2 = function () {
  var emsg$0 = undefined;var i$5 = undefined;var n$3 = undefined;var values$3 = undefined;var pos$0 = undefined;var t0$15 = undefined;var m$15$0 = undefined;var __at___$18 = undefined;var self$21 = undefined;__at___$18 = this;self$21 = this;m$15$0 = arguments;t0$15 = m$15$0.length;if (t0$15 >= 3 && t0$15 <= 4) {
    i$5 = m$15$0[0];n$3 = m$15$0[1];values$3 = m$15$0[2];if (3 >= t0$15) {
      pos$0 = false;
    } else {
      pos$0 = m$15$0[3];
    }emsg$0 = "Inconsistent splicing (this may happen if you modify" + " the array more than once at the same place)";return send(self$21, symbols$0.registerPatchFunction, true)("@splice", function (d$1) {
      var $targ$12 = undefined;var m$17$0 = undefined;var new__minus__values$0 = undefined;var $targ$13 = undefined;var n2$0 = undefined;var previous__minus__values$0 = undefined;var t0$16 = undefined;var m$16$0 = undefined;var m$18 = undefined;d$1 = d$1 || Array(send(self$21, symbols$0.obj).length + 1);m$16$0 = send(d$1, i$5);if (m$16$0 === true) {
        throw ErrorFactory(["splice"]).create(emsg$0);
      } else {
        if (m$16$0 === void 0) {
          $targ$12 = [n$3, values$3];d$1[i$5] = $targ$12;void 0;
        } else {
          if (Array.isArray(m$16$0) && (t0$16 = m$16$0.length, t0$16 === 2 && (n2$0 = m$16$0[0], previous__minus__values$0 = m$16$0[1], equal(n$3, 0) || equal(n2$0, 0)))) {
            m$17$0 = pos$0;if (m$17$0 === "prepend") {
              new__minus__values$0 = values$3.concat(previous__minus__values$0);
            } else {
              if (m$17$0 === "append") {
                new__minus__values$0 = previous__minus__values$0.concat(values$3);
              } else {
                throw ErrorFactory(["splice"]).create(emsg$0);
              }
            }$targ$13 = [Math.max(n$3, n2$0), new__minus__values$0];d$1[i$5] = $targ$13;void 0;
          } else {
            throw ErrorFactory(["splice"]).create(emsg$0);
          }
        }
      }m$18 = null;$11: for (var _iterator = range(1, n$3 - 1)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
        m$18 = _step.value;
        var $targ$14 = undefined;var m$19$0 = undefined;var j$2 = undefined;j$2 = m$18;m$19$0 = send(d$1, i$5 + j$2);if (m$19$0 === void 0) {
          $targ$14 = true;d$1[i$5 + j$2] = $targ$14;void 0;
        } else {
          throw ErrorFactory(["splice"]).create(emsg$0);
        }
      }return d$1;
    });
  } else {
    return ___match_error(m$15$0, "{i, n, values, pos = false}");
  }
};array__minus__proxy$0.prototype._splice = $targ$2;__amp____colon__(array__minus__proxy$0, __amp____colon__(($targ$15 = "array-proxy", accum$3 = {}, accum$3["::name"] = $targ$15, accum$3), ($targ$16 = true, accum$4 = {}, accum$4["::egclass"] = $targ$16, accum$4)));Object.defineProperty(array__minus__proxy$0.prototype, "length", ($targ$17 = function () {
  send(this, symbols$0.registerTracks, true)("length");return proxy$0(send(this, symbols$0.obj).length, send(this, symbols$0.tracks), send(this, symbols$0.patch));
}, accum$5 = {}, accum$5.get = $targ$17, accum$5));$targ$18 = proxy$0;exports.proxy = $targ$18;$targ$19 = deprox$0;exports.deprox = $targ$19;$targ$20 = reprox$0;exports.reprox = $targ$20;$targ$21 = object__minus__proxy$0;exports.objectProxy = $targ$21;$targ$22 = array__minus__proxy$0;exports.arrayProxy = $targ$22;$targ$23 = read__minus__proxy$0;exports.readProxy = $targ$23;$targ$24 = write__minus__proxy$0;exports.writeProxy = $targ$24;$targ$25 = setup__minus__proxy$0;exports.setupProxy = $targ$25;void 0;
