// quasar/lib/state.js
"use strict";

require("earlgrey-runtime/5");var $targ$0 = undefined;var accum$0 = undefined;var $targ$1 = undefined;var accum$1 = undefined;var $targ$14 = undefined;var accum$2 = undefined;var $targ$15 = undefined;var accum$3 = undefined;var $targ$29 = undefined;var accum$4 = undefined;var $targ$30 = undefined;var accum$5 = undefined;var $targ$31 = undefined;var $targ$32 = undefined;var $0$0 = undefined;var proxy$0 = undefined;var directory$0 = undefined;var Directory$0 = undefined;var tracks$0 = undefined;var Tracks$0 = undefined;var Patch$0 = undefined;var util$0 = undefined;var symbols$0 = undefined;var cache$0 = undefined;var Frame$0 = undefined;var with__minus__frame$0 = undefined;var events$0 = undefined;var EventEmitter$0 = undefined;var EE$0 = undefined;var State$0 = undefined;var UNCOMPUTED$0 = undefined;var Function = undefined;$0$0 = require("./proxy");proxy$0 = getProperty($0$0, "proxy", "./proxy");directory$0 = require("./directory");Directory$0 = getProperty(directory$0, "Directory", "directory");tracks$0 = require("./tracks");Tracks$0 = getProperty(tracks$0, "Tracks", "tracks");Patch$0 = getProperty(tracks$0, "Patch", "tracks");util$0 = require("./util");symbols$0 = getProperty(util$0, "symbols", "util");cache$0 = require("./cache");Frame$0 = getProperty(cache$0, "Frame", "cache");with__minus__frame$0 = getProperty(cache$0, "withFrame", "cache");events$0 = require("events");EventEmitter$0 = getProperty(events$0, "EventEmitter", "events");EE$0 = ___extend(function () {
  var t0$0 = undefined;var m$0$0 = undefined;var __at___$0 = undefined;if (!getChecker(EE$0)(this)) {
    __at___$0 = Object.create(EE$0.prototype);
  } else {
    __at___$0 = this;
  }m$0$0 = arguments;t0$0 = m$0$0.length;if (t0$0 >= 0 && t0$0 <= 1) {
    __at___$0._self = 0 >= t0$0 ? null : m$0$0[0];EventEmitter$0.constructor.call(__at___$0);
  } else {
    ___match_error(m$0$0, "{@_self = null}");
  }return __at___$0;
}, EventEmitter$0);EE$0.prototype.emit = function emit() {
  var event$0 = undefined;var args$0 = undefined;var t0$1 = undefined;var m$1$0 = undefined;var __at___$1 = undefined;var self$0 = undefined;__at___$1 = this;self$0 = this;m$1$0 = arguments;t0$1 = m$1$0.length;if (t0$1 >= 1) {
    event$0 = m$1$0[0];args$0 = Array.prototype.slice.call(m$1$0, 1);return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
      var m$2, acc$0, temp$0, _iterator, _step, x$0, p$0, m$3$0, listener$0;

      return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
        while (1) switch (context$2$0.prev = context$2$0.next) {
          case 0:
            m$2 = undefined;
            acc$0 = undefined;
            temp$0 = undefined;
            acc$0 = [];temp$0 = null;m$2 = null;_iterator = __at___$1.listeners(event$0)[Symbol.iterator]();

          case 7:
            if ((_step = _iterator.next()).done) {
              context$2$0.next = 27;
              break;
            }

            m$2 = _step.value;
            x$0 = undefined;
            p$0 = undefined;
            m$3$0 = undefined;
            listener$0 = undefined;
            listener$0 = m$2;m$3$0 = listener$0.apply(__at___$1._self, args$0);
            if (!getChecker(Promise)(m$3$0)) {
              context$2$0.next = 22;
              break;
            }

            p$0 = m$3$0;context$2$0.next = 19;
            return p$0;

          case 19:
            temp$0 = context$2$0.sent;
            context$2$0.next = 24;
            break;

          case 22:
            x$0 = m$3$0;temp$0 = x$0;

          case 24:
            acc$0.push(temp$0);

          case 25:
            context$2$0.next = 7;
            break;

          case 27:
            return context$2$0.abrupt("return", acc$0);

          case 28:
          case "end":
            return context$2$0.stop();
        }
      }, callee$1$0, this);
    }), true);
  } else {
    return ___match_error(m$1$0, "{event, *args}");
  }
};__amp____colon__(EE$0, __amp____colon__(($targ$0 = "EE", accum$0 = {}, accum$0["::name"] = $targ$0, accum$0), ($targ$1 = true, accum$1 = {}, accum$1["::egclass"] = $targ$1, accum$1)));State$0 = function State() {
  var t0$3 = undefined;var m$5 = undefined;var $targ$2 = undefined;var $targ$3 = undefined;var $targ$4 = undefined;var $targ$5 = undefined;var $targ$6 = undefined;var $targ$7 = undefined;var $targ$8 = undefined;var $targ$9 = undefined;var $targ$10 = undefined;var t0$2 = undefined;var m$4$0 = undefined;var __at___$2 = undefined;if (!getChecker(State$0)(this)) {
    __at___$2 = Object.create(State$0.prototype);
  } else {
    __at___$2 = this;
  }m$4$0 = arguments;t0$2 = m$4$0.length;if (t0$2 >= 1 && t0$2 <= 2) {
    __at___$2.origin = m$4$0[0];__at___$2.options = 1 >= t0$2 ? {} : m$4$0[1];$targ$2 = __at___$2.options;t0$3 = $targ$2;__at___$2.clobberPatch = ___hasprop(t0$3, "clobberPatch") ? t0$3.clobberPatch : false;__at___$2.inputSelf = ___hasprop(t0$3, "inputSelf") ? t0$3.inputSelf : false;__at___$2.commitPolicy = ___hasprop(t0$3, "commitPolicy") ? t0$3.commitPolicy : "transact";$targ$3 = new EE$0();__at___$2.emitter = $targ$3;m$5 = null;$2: for (var _iterator = items(__at___$2.options.listeners || {})[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      m$5 = _step.value;
      var ename$0 = undefined;var fn$0 = undefined;var t0$4 = undefined;var t1$0 = undefined;t0$4 = m$5;if (Array.isArray(t0$4) && (t1$0 = t0$4.length, t1$0 === 2)) {
        ename$0 = t0$4[0];fn$0 = t0$4[1];__at___$2.emitter.on(ename$0, fn$0);
      } else {
        ___match_error(m$5);
      }
    }__at___$2.emitter.on("newListener", function (temp$1$0, listener$1) {
      var t0$5 = undefined;var event$1 = undefined;var ph$0$0 = undefined;t0$5 = temp$1$0;event$1 = t0$5;ph$0$0 = t0$5;if (ph$0$0 === "state") {
        return listener$1(__at___$2.model());
      } else {
        return undefined;
      }
    });$targ$4 = 0;__at___$2.ntrans = $targ$4;$targ$5 = Patch$0();__at___$2.patch = $targ$5;$targ$6 = send(Directory$0, { clobberPatch: __at___$2.clobberPatch });__at___$2.dir = $targ$6;$targ$7 = __at___$2;__at___$2.dir[symbols$0.associatedState] = $targ$7;$targ$8 = __at___$2.dir.acquire(__at___$2.origin);__at___$2.basis = $targ$8;$targ$9 = proxy$0(__at___$2.basis, null, __at___$2.patch);__at___$2.basisProxy = $targ$9;__at___$2.patch.shift();$targ$10 = false;__at___$2.outstandingCommit = $targ$10;void 0;
  } else {
    ___match_error(m$4$0, "{@origin, @options = {=}}");
  }return __at___$2;
};State$0.prototype.on = function on(event$2, listener$2) {
  var __at___$3 = undefined;var self$1 = undefined;__at___$3 = this;self$1 = this;return __at___$3.emitter.on(event$2, listener$2);
};State$0.prototype.sync = function sync(listener$3) {
  var __at___$4 = undefined;var self$2 = undefined;__at___$4 = this;self$2 = this;return __at___$4.on("state", listener$3);
};State$0.prototype.get = function get() {
  var __at___$5 = undefined;var self$3 = undefined;__at___$5 = this;self$3 = this;return __at___$5.basis;
};State$0.prototype.self = function self() {
  var __at___$6 = undefined;var self$4 = undefined;__at___$6 = this;self$4 = this;if (__at___$6.inputSelf) {
    return __at___$6;
  } else {
    return __at___$6.basis;
  }
};State$0.prototype.model = function model() {
  var v$0 = undefined;var t0$6 = undefined;var m$6$0 = undefined;var __at___$7 = undefined;var self$5 = undefined;__at___$7 = this;self$5 = this;m$6$0 = arguments;t0$6 = m$6$0.length;if (t0$6 >= 0 && t0$6 <= 1) {
    if (0 >= t0$6) {
      v$0 = null;
    } else {
      v$0 = m$6$0[0];
    }if (v$0 === null) {
      return __at___$7.basisProxy;
    } else {
      return proxy$0(v$0, null, __at___$7.patch);
    }
  } else {
    return ___match_error(m$6$0, "{v = null}");
  }
};State$0.prototype.commit = function commit() {
  var __at___$8 = undefined;var self$6 = undefined;__at___$8 = this;self$6 = this;return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
    var $targ$11, t0$7, x$1, p$1, m$7$0, x$2, p$2, m$8$0, $targ$12, patch$0, $targ$13;
    return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
      while (1) switch (context$2$0.prev = context$2$0.next) {
        case 0:
          $targ$11 = undefined;
          t0$7 = undefined;
          x$1 = undefined;
          p$1 = undefined;
          m$7$0 = undefined;
          x$2 = undefined;
          p$2 = undefined;
          m$8$0 = undefined;
          $targ$12 = undefined;
          patch$0 = undefined;
          $targ$13 = undefined;

          if (!(__at___$8.ntrans > 0 || __at___$8.committing)) {
            context$2$0.next = 17;
            break;
          }

          $targ$11 = true;__at___$8.outstandingCommit = $targ$11;return context$2$0.abrupt("return", false);

        case 17:
          $targ$12 = true;__at___$8.committing = $targ$12;patch$0 = Patch$0(__at___$8.patch.shift());__at___$8.dir.patch(patch$0);$targ$13 = false;t0$7 = $targ$13;__at___$8.outstandingCommit = t0$7;__at___$8.committing = t0$7;m$7$0 = __at___$8.emitter.emit("commit", __at___$8.model(), patch$0);
          if (!getChecker(Promise)(m$7$0)) {
            context$2$0.next = 32;
            break;
          }

          p$1 = m$7$0;context$2$0.next = 30;
          return p$1;

        case 30:
          context$2$0.next = 34;
          break;

        case 32:
          x$1 = m$7$0;x$1;

        case 34:
          m$8$0 = __at___$8.emitter.emit("state", __at___$8.model(), patch$0);
          if (!getChecker(Promise)(m$8$0)) {
            context$2$0.next = 41;
            break;
          }

          p$2 = m$8$0;context$2$0.next = 39;
          return p$2;

        case 39:
          context$2$0.next = 43;
          break;

        case 41:
          x$2 = m$8$0;x$2;

        case 43:
          return context$2$0.abrupt("return", true);

        case 44:
        case "end":
          return context$2$0.stop();
      }
    }, callee$1$0, this);
  }), true);
};State$0.prototype.transact = function transact(fn$1) {
  var __at___$9 = undefined;var self$7 = undefined;__at___$9 = this;self$7 = this;return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
    var x$3, p$3, m$9$0;
    return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
      while (1) switch (context$2$0.prev = context$2$0.next) {
        case 0:
          x$3 = undefined;
          p$3 = undefined;
          m$9$0 = undefined;
          __at___$9.ntrans = __at___$9.ntrans + 1;m$9$0 = fn$1(__at___$9.model());
          if (!getChecker(Promise)(m$9$0)) {
            context$2$0.next = 11;
            break;
          }

          p$3 = m$9$0;context$2$0.next = 9;
          return p$3;

        case 9:
          context$2$0.next = 13;
          break;

        case 11:
          x$3 = m$9$0;x$3;

        case 13:
          __at___$9.ntrans = __at___$9.ntrans - 1;
          if (!(__at___$9.ntrans === 0 && (__at___$9.commitPolicy === "transact" || __at___$9.outstandingCommit))) {
            context$2$0.next = 18;
            break;
          }

          return context$2$0.abrupt("return", __at___$9.commit());

        case 18:
          return context$2$0.abrupt("return", __at___$9.emitter.emit("transact", __at___$9));

        case 19:
        case "end":
          return context$2$0.stop();
      }
    }, callee$1$0, this);
  }), true);
};__amp____colon__(State$0, __amp____colon__(($targ$14 = "State", accum$2 = {}, accum$2["::name"] = $targ$14, accum$2), ($targ$15 = true, accum$3 = {}, accum$3["::egclass"] = $targ$15, accum$3)));UNCOMPUTED$0 = Symbol("UNCOMPUTED");Function = function Function() {
  var m$11 = undefined;var t0$11 = undefined;var $targ$16 = undefined;var $targ$17 = undefined;var $targ$18 = undefined;var $targ$19 = undefined;var t0$8 = undefined;var m$10$0 = undefined;var __at___$10 = undefined;if (!(typeof this === "function")) {
    __at___$10 = Object.create(Function.prototype);
  } else {
    __at___$10 = this;
  }m$10$0 = arguments;t0$8 = m$10$0.length;if (t0$8 >= 2 && t0$8 <= 3) {
    __at___$10.input = m$10$0[0];__at___$10.render = m$10$0[1];__at___$10.options = 2 >= t0$8 ? {} : m$10$0[2];$targ$16 = EE$0();__at___$10.emitter = $targ$16;m$11 = null;$3: for (var _iterator = items(__at___$10.options.listeners || {})[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      m$11 = _step.value;
      var ename$1 = undefined;var fn$2 = undefined;var t0$9 = undefined;var t1$1 = undefined;t0$9 = m$11;if (Array.isArray(t0$9) && (t1$1 = t0$9.length, t1$1 === 2)) {
        ename$1 = t0$9[0];fn$2 = t0$9[1];__at___$10.emitter.on(ename$1, fn$2);
      } else {
        ___match_error(m$11);
      }
    }__at___$10.emitter.on("newListener", function (temp$2$0, listener$4) {
      var t0$10 = undefined;var event$3 = undefined;var ph$1$0 = undefined;t0$10 = temp$2$0;event$3 = t0$10;ph$1$0 = t0$10;if (ph$1$0 === "value") {
        return spawn.call(this, regeneratorRuntime.mark(function callee$2$0() {
          var rval$0, e$0;
          return regeneratorRuntime.wrap(function callee$2$0$(context$3$0) {
            while (1) switch (context$3$0.prev = context$3$0.next) {
              case 0:
                rval$0 = undefined;
                rval$0 = false;context$3$0.prev = 2;

                if (!(__at___$10.value !== UNCOMPUTED$0)) {
                  context$3$0.next = 10;
                  break;
                }

                context$3$0.next = 6;
                return __at___$10.value;

              case 6:
                context$3$0.t4 = context$3$0.sent;
                context$3$0.t3 = listener$4(context$3$0.t4);
                context$3$0.next = 11;
                break;

              case 10:
                context$3$0.t3 = undefined;

              case 11:
                context$3$0.next = 13;
                return context$3$0.t3;

              case 13:
                rval$0 = context$3$0.sent;
                rval$0;context$3$0.next = 23;
                break;

              case 17:
                context$3$0.prev = 17;
                context$3$0.t5 = context$3$0["catch"](2);
                e$0 = undefined;
                e$0 = context$3$0.t5;rval$0 = console.error(e$0.stack);rval$0;

              case 23:
                return context$3$0.abrupt("return", rval$0);

              case 24:
              case "end":
                return context$3$0.stop();
            }
          }, callee$2$0, this, [[2, 17]]);
        }), true);
      } else {
        if (ph$1$0 === "effects") {
          return spawn.call(this, regeneratorRuntime.mark(function callee$2$1() {
            var rval$1, e$1;
            return regeneratorRuntime.wrap(function callee$2$1$(context$3$0) {
              while (1) switch (context$3$0.prev = context$3$0.next) {
                case 0:
                  rval$1 = undefined;
                  rval$1 = false;context$3$0.prev = 2;

                  if (!(__at___$10.value !== UNCOMPUTED$0)) {
                    context$3$0.next = 9;
                    break;
                  }

                  context$3$0.next = 6;
                  return __at___$10.value;

                case 6:
                  context$3$0.t6 = listener$4(__at___$10.basis.effects);
                  context$3$0.next = 10;
                  break;

                case 9:
                  context$3$0.t6 = undefined;

                case 10:
                  context$3$0.next = 12;
                  return context$3$0.t6;

                case 12:
                  rval$1 = context$3$0.sent;
                  rval$1;context$3$0.next = 22;
                  break;

                case 16:
                  context$3$0.prev = 16;
                  context$3$0.t7 = context$3$0["catch"](2);
                  e$1 = undefined;
                  e$1 = context$3$0.t7;rval$1 = console.error(e$1.stack);rval$1;

                case 22:
                  return context$3$0.abrupt("return", rval$1);

                case 23:
                case "end":
                  return context$3$0.stop();
              }
            }, callee$2$1, this, [[2, 16]]);
          }), true);
        } else {
          if (ph$1$0 === "error") {
            if (__at___$10.error) {
              return listener$4(__at___$10.error);
            }
          } else {
            return undefined;
          }
        }
      }
    });$targ$17 = __at___$10.options;t0$11 = $targ$17;__at___$10.clobberPatch = ___hasprop(t0$11, "clobberPatch") ? t0$11.clobberPatch : true;__at___$10.sideEffects = ___hasprop(t0$11, "sideEffects") ? t0$11.sideEffects : false;$targ$18 = UNCOMPUTED$0;__at___$10.value = $targ$18;$targ$19 = true;__at___$10.redo = $targ$19;__at___$10.input.sync(__at___$10.recompute.bind(__at___$10));
  } else {
    ___match_error(m$10$0, "{@input, @render, @options = {=}}");
  }return __at___$10;
};Function.prototype.compute = function compute() {
  var patch$1 = undefined;var t0$12 = undefined;var m$12$0 = undefined;var __at___$11 = undefined;var self$8 = undefined;__at___$11 = this;self$8 = this;m$12$0 = arguments;t0$12 = m$12$0.length;if (t0$12 >= 0 && t0$12 <= 1) {
    if (0 >= t0$12) {
      patch$1 = null;
    } else {
      patch$1 = m$12$0[0];
    }return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
      var $targ$22, x$6, p$6, m$13$0, context$0, fn$3, $targ$20, $targ$21;
      return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
        while (1) switch (context$2$0.prev = context$2$0.next) {
          case 0:
            $targ$22 = undefined;
            x$6 = undefined;
            p$6 = undefined;
            m$13$0 = undefined;
            context$0 = undefined;
            fn$3 = undefined;
            $targ$20 = undefined;
            $targ$21 = undefined;
            if (__at___$11.sideEffects || !__at___$11.redo && __at___$11._previousInput !== __at___$11.input.get()) {
              $targ$22 = true;__at___$11.redo = $targ$22;void 0;
            }$targ$20 = __at___$11.input.get();__at___$11._previousInput = $targ$20;context$0 = { events: __at___$11.emitter, state: __at___$11.input, patch: patch$1, dir: __at___$11.input.dir, outPatch: __at___$11.patch, root: true };fn$3 = function fn() {
              return spawn.call(this, regeneratorRuntime.mark(function callee$3$0() {
                var x$4, p$4, m$14$0, $targ$23, x$5, p$5, m$15$0, $targ$24;
                return regeneratorRuntime.wrap(function callee$3$0$(context$4$0) {
                  while (1) switch (context$4$0.prev = context$4$0.next) {
                    case 0:
                      x$4 = undefined;
                      p$4 = undefined;
                      m$14$0 = undefined;
                      $targ$23 = undefined;
                      x$5 = undefined;
                      p$5 = undefined;
                      m$15$0 = undefined;
                      $targ$24 = undefined;

                      if (!__at___$11.redo) {
                        context$4$0.next = 23;
                        break;
                      }

                      $targ$23 = false;__at___$11.redo = $targ$23;m$14$0 = __at___$11.render.rend.call(null, __at___$11.input.model());
                      if (!getChecker(Promise)(m$14$0)) {
                        context$4$0.next = 19;
                        break;
                      }

                      p$4 = m$14$0;context$4$0.next = 16;
                      return p$4;

                    case 16:
                      return context$4$0.abrupt("return", context$4$0.sent);

                    case 19:
                      x$4 = m$14$0;return context$4$0.abrupt("return", x$4);

                    case 21:
                      context$4$0.next = 35;
                      break;

                    case 23:
                      m$15$0 = __at___$11.basis.update();
                      if (!getChecker(Promise)(m$15$0)) {
                        context$4$0.next = 31;
                        break;
                      }

                      p$5 = m$15$0;context$4$0.next = 28;
                      return p$5;

                    case 28:
                      $targ$24 = context$4$0.sent;
                      context$4$0.next = 33;
                      break;

                    case 31:
                      x$5 = m$15$0;$targ$24 = x$5;

                    case 33:
                      __at___$11.basis = $targ$24;return context$4$0.abrupt("return", __at___$11.basis);

                    case 35:
                    case "end":
                      return context$4$0.stop();
                  }
                }, callee$3$0, this);
              }), true);
            };m$13$0 = with__minus__frame$0(Frame$0(Tracks$0(), [], context$0), fn$3);
            if (!getChecker(Promise)(m$13$0)) {
              context$2$0.next = 21;
              break;
            }

            p$6 = m$13$0;context$2$0.next = 18;
            return p$6;

          case 18:
            $targ$21 = context$2$0.sent;
            context$2$0.next = 23;
            break;

          case 21:
            x$6 = m$13$0;$targ$21 = x$6;

          case 23:
            __at___$11.basis = $targ$21;return context$2$0.abrupt("return", __at___$11.basis.value);

          case 25:
          case "end":
            return context$2$0.stop();
        }
      }, callee$1$0, this);
    }), true);
  } else {
    return ___match_error(m$12$0, "{patch = null}");
  }
};Function.prototype.recompute = function recompute(model$0, patch$2) {
  var __at___$12 = undefined;var self$9 = undefined;__at___$12 = this;self$9 = this;return spawn.call(this, regeneratorRuntime.mark(function callee$1$0() {
    var x$7, p$7, m$17$0, x$8, p$8, m$16$0, $targ$25, $targ$26, rval$2, x$9, p$9, m$18$0, $targ$27, $targ$28, e$2;
    return regeneratorRuntime.wrap(function callee$1$0$(context$2$0) {
      while (1) switch (context$2$0.prev = context$2$0.next) {
        case 0:
          x$7 = undefined;
          p$7 = undefined;
          m$17$0 = undefined;
          x$8 = undefined;
          p$8 = undefined;
          m$16$0 = undefined;
          $targ$25 = undefined;
          $targ$26 = undefined;
          rval$2 = undefined;
          rval$2 = false;context$2$0.prev = 10;
          $targ$25 = null;__at___$12.error = $targ$25;$targ$26 = __at___$12.compute(patch$2);__at___$12.value = $targ$26;m$17$0 = __at___$12.value;

          if (!getChecker(Promise)(m$17$0)) {
            context$2$0.next = 23;
            break;
          }

          p$7 = m$17$0;
          context$2$0.next = 20;
          return p$7;

        case 20:
          context$2$0.t8 = context$2$0.sent;
          context$2$0.next = 24;
          break;

        case 23:
          context$2$0.t8 = (x$7 = m$17$0, x$7);

        case 24:
          m$16$0 = __at___$12.emitter.emit("value", context$2$0.t8);

          if (!getChecker(Promise)(m$16$0)) {
            context$2$0.next = 31;
            break;
          }

          p$8 = m$16$0;context$2$0.next = 29;
          return p$8;

        case 29:
          context$2$0.next = 33;
          break;

        case 31:
          x$8 = m$16$0;x$8;

        case 33:
          rval$2 = __at___$12.emitter.emit("effects", __at___$12.basis.effects);rval$2;context$2$0.next = 61;
          break;

        case 37:
          context$2$0.prev = 37;
          context$2$0.t9 = context$2$0["catch"](10);
          x$9 = undefined;
          p$9 = undefined;
          m$18$0 = undefined;
          $targ$27 = undefined;
          $targ$28 = undefined;
          e$2 = undefined;
          e$2 = context$2$0.t9;$targ$27 = e$2;__at___$12.error = $targ$27;$targ$28 = true;__at___$12.redo = $targ$28;m$18$0 = __at___$12.emitter.emit("error", e$2);
          if (!getChecker(Promise)(m$18$0)) {
            context$2$0.next = 58;
            break;
          }

          p$9 = m$18$0;context$2$0.next = 55;
          return p$9;

        case 55:
          rval$2 = context$2$0.sent;
          context$2$0.next = 60;
          break;

        case 58:
          x$9 = m$18$0;rval$2 = x$9;

        case 60:
          rval$2;

        case 61:
          return context$2$0.abrupt("return", rval$2);

        case 62:
        case "end":
          return context$2$0.stop();
      }
    }, callee$1$0, this, [[10, 37]]);
  }), true);
};Function.prototype.get = function get() {
  var __at___$13 = undefined;var self$10 = undefined;__at___$13 = this;self$10 = this;if (__at___$13.error) {
    throw __at___$13.error;
  }return __at___$13.value;
};Function.prototype.sync = function sync(fn$4) {
  var __at___$14 = undefined;var self$11 = undefined;__at___$14 = this;self$11 = this;return __at___$14.emitter.on("value", fn$4);
};Function.prototype.syncEffects = function (fn$5) {
  var __at___$15 = undefined;var self$12 = undefined;__at___$15 = this;self$12 = this;return __at___$15.emitter.on("effects", fn$5);
};Function.prototype.on = function on(event$4, fn$6) {
  var __at___$16 = undefined;var self$13 = undefined;__at___$16 = this;self$13 = this;return __at___$16.emitter.on(event$4, fn$6);
};__amp____colon__(Function, __amp____colon__(($targ$29 = "Function", accum$4 = {}, accum$4["::name"] = $targ$29, accum$4), ($targ$30 = true, accum$5 = {}, accum$5["::egclass"] = $targ$30, accum$5)));$targ$31 = State$0;exports.State = $targ$31;$targ$32 = Function;exports.Function = $targ$32;void 0;
