// quasar/lib/tracks.js
"use strict";

require("earlgrey-runtime/5");var $targ$2 = undefined;var accum$0 = undefined;var $targ$3 = undefined;var accum$1 = undefined;var $targ$4 = undefined;var accum$2 = undefined;var $targ$5 = undefined;var accum$3 = undefined;var $targ$7 = undefined;var accum$4 = undefined;var $targ$8 = undefined;var accum$5 = undefined;var $targ$11 = undefined;var accum$6 = undefined;var $targ$12 = undefined;var accum$7 = undefined;var $targ$14 = undefined;var accum$8 = undefined;var $targ$15 = undefined;var accum$9 = undefined;var $targ$18 = undefined;var accum$10 = undefined;var $targ$19 = undefined;var accum$11 = undefined;var $targ$20 = undefined;var $targ$21 = undefined;var $targ$22 = undefined;var $targ$23 = undefined;var $targ$24 = undefined;var $targ$25 = undefined;var util$0 = undefined;var ID$0 = undefined;var symbols$0 = undefined;var ObjectTrack$0 = undefined;var $targ$0 = undefined;var ArrayTrack$0 = undefined;var $targ$1 = undefined;var Counter$0 = undefined;var Chirurgies$0 = undefined;var Tracks$0 = undefined;var Patch$0 = undefined;util$0 = require("./util");ID$0 = getProperty(util$0, "ID", "util");symbols$0 = getProperty(util$0, "symbols", "util");ObjectTrack$0 = function ObjectTrack(init$0) {
  var __at___$0 = undefined;if (!getChecker(ObjectTrack$0)(this)) {
    __at___$0 = Object.create(ObjectTrack$0.prototype);
  } else {
    __at___$0 = this;
  }init$0 && __amp____colon__(__at___$0, init$0);return __at___$0;
};$targ$0 = regeneratorRuntime.mark(function callee$0$0(writes$0) {
  var _this = this;

  var m$0, acc$0, temp$0, __at___$1, self$0, _iterator, _step, tmp$0, name$0, r$0, t0$0, t1$0;

  return regeneratorRuntime.wrap(function callee$0$0$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        m$0 = undefined;
        acc$0 = undefined;
        temp$0 = undefined;
        __at___$1 = undefined;
        self$0 = undefined;
        __at___$1 = _this;self$0 = _this;acc$0 = [];temp$0 = null;m$0 = null;_iterator = items(writes$0 || {})[Symbol.iterator]();

      case 11:
        if ((_step = _iterator.next()).done) {
          context$1$0.next = 29;
          break;
        }

        m$0 = _step.value;
        tmp$0 = undefined;
        name$0 = undefined;
        r$0 = undefined;
        t0$0 = undefined;
        t1$0 = undefined;
        t0$0 = m$0;
        if (!(Array.isArray(t0$0) && (t1$0 = t0$0.length, t1$0 === 2 && (name$0 = t0$0[0], t0$0[1], r$0 = send(__at___$1, name$0), tmp$0 = r$0, tmp$0 === undefined || tmp$0 === null ? tmp$0 : tmp$0.notEmpty())))) {
          context$1$0.next = 26;
          break;
        }

        context$1$0.next = 22;
        return r$0;

      case 22:
        temp$0 = context$1$0.sent;
        acc$0.push(temp$0);context$1$0.next = 27;
        break;

      case 26:
        false;

      case 27:
        context$1$0.next = 11;
        break;

      case 29:
        return context$1$0.abrupt("return", acc$0);

      case 30:
      case "end":
        return context$1$0.stop();
    }
  }, callee$0$0, this);
});ObjectTrack$0.prototype[symbols$0.relevantChanges] = $targ$0;__amp____colon__(ObjectTrack$0, __amp____colon__(($targ$2 = "ObjectTrack", accum$0 = {}, accum$0["::name"] = $targ$2, accum$0), ($targ$3 = true, accum$1 = {}, accum$1["::egclass"] = $targ$3, accum$1)));ArrayTrack$0 = function ArrayTrack(init$1) {
  var __at___$2 = undefined;if (!getChecker(ArrayTrack$0)(this)) {
    __at___$2 = Object.create(ArrayTrack$0.prototype);
  } else {
    __at___$2 = this;
  }init$1 && __amp____colon__(__at___$2, init$1);return __at___$2;
};$targ$1 = regeneratorRuntime.mark(function callee$0$1(writes$1) {
  var _this = this;

  var m$2, acc$1, temp$1, m$3, acc$2, temp$2, mask$0, taints$0, splices$0, m$1$0, __at___$3, self$1, _iterator, _step, t0$1, t1$1, t2$0, t3$0, _iterator2, _step2, tmp$1, n$0, v$1, v$0, $$6965$0, $$6966$0, $$6967$0, $$6968$0, t0$2, t1$2;

  return regeneratorRuntime.wrap(function callee$0$1$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        m$2 = undefined;
        acc$1 = undefined;
        temp$1 = undefined;
        m$3 = undefined;
        acc$2 = undefined;
        temp$2 = undefined;
        mask$0 = undefined;
        taints$0 = undefined;
        splices$0 = undefined;
        m$1$0 = undefined;
        __at___$3 = undefined;
        self$1 = undefined;
        __at___$3 = _this;self$1 = _this;m$1$0 = writes$1["@splice"];
        if (!(m$1$0 === void 0)) {
          context$1$0.next = 19;
          break;
        }

        return context$1$0.abrupt("return", undefined);

      case 19:
        splices$0 = m$1$0;mask$0 = 3;acc$1 = [];temp$1 = null;m$2 = null;$2: for (_iterator = splices$0[Symbol.iterator](); !(_step = _iterator.next()).done;) {
          m$2 = _step.value;
          t0$1 = undefined;
          t1$1 = undefined;
          t2$0 = undefined;
          t3$0 = undefined;
          if (m$2 === void 0) {
            temp$1 = 3 & mask$0;acc$1.push(temp$1);
          } else {
            t0$1 = m$2;if (Array.isArray(t0$1) && (t1$1 = t0$1.length, t1$1 === 2 && (t0$1[0] === 1 && (t2$0 = t0$1[1], Array.isArray(t2$0) && (t3$0 = t2$0.length, t3$0 === 1))))) {
              t2$0[0];temp$1 = 1 & mask$0;acc$1.push(temp$1);
            } else {
              mask$0 = 0;temp$1 = 0;acc$1.push(temp$1);
            }
          }
        }taints$0 = acc$1;acc$2 = [];temp$2 = null;m$3 = null;_iterator2 = items(__at___$3)[Symbol.iterator]();

      case 30:
        if ((_step2 = _iterator2.next()).done) {
          context$1$0.next = 62;
          break;
        }

        m$3 = _step2.value;
        tmp$1 = undefined;
        n$0 = undefined;
        v$1 = undefined;
        v$0 = undefined;
        $$6965$0 = undefined;
        $$6966$0 = undefined;
        $$6967$0 = undefined;
        $$6968$0 = undefined;
        t0$2 = undefined;
        t1$2 = undefined;
        t0$2 = m$3;
        if (!(($$6966$0 = Array.isArray(t0$2)) && (t1$2 = t0$2.length, ($$6968$0 = t1$2 === 2) && t0$2[0] === "@iter"))) {
          context$1$0.next = 51;
          break;
        }

        v$0 = t0$2[1];context$1$0.next = 47;
        return v$0;

      case 47:
        temp$2 = context$1$0.sent;
        acc$2.push(temp$2);context$1$0.next = 60;
        break;

      case 51:
        if (!($$6968$0 && (n$0 = t0$2[0], v$1 = t0$2[1], tmp$1 = v$1, tmp$1 === undefined || tmp$1 === null ? tmp$1 : tmp$1.notEmpty()))) {
          context$1$0.next = 59;
          break;
        }

        if (!nequal(send(taints$0, n$0), 3)) {
          context$1$0.next = 56;
          break;
        }

        context$1$0.next = 55;
        return v$1;

      case 55:
        temp$2 = context$1$0.sent;

      case 56:
        acc$2.push(temp$2);context$1$0.next = 60;
        break;

      case 59:
        false;

      case 60:
        context$1$0.next = 30;
        break;

      case 62:
        return context$1$0.abrupt("return", acc$2);

      case 63:
      case "end":
        return context$1$0.stop();
    }
  }, callee$0$1, this);
});ArrayTrack$0.prototype[symbols$0.relevantChanges] = $targ$1;__amp____colon__(ArrayTrack$0, __amp____colon__(($targ$4 = "ArrayTrack", accum$2 = {}, accum$2["::name"] = $targ$4, accum$2), ($targ$5 = true, accum$3 = {}, accum$3["::egclass"] = $targ$5, accum$3)));Counter$0 = function Counter() {
  var $targ$6 = undefined;var __at___$4 = undefined;if (!getChecker(Counter$0)(this)) {
    __at___$4 = Object.create(Counter$0.prototype);
  } else {
    __at___$4 = this;
  }$targ$6 = 0;__at___$4.nTracks = $targ$6;return __at___$4;
};Counter$0.prototype.mark = function mark() {
  var __at___$5 = undefined;var self$2 = undefined;__at___$5 = this;self$2 = this;return __at___$5.nTracks = __at___$5.nTracks + 1;
};Counter$0.prototype.unmark = function unmark() {
  var __at___$6 = undefined;var self$3 = undefined;__at___$6 = this;self$3 = this;return __at___$6.nTracks = __at___$6.nTracks - 1;
};Counter$0.prototype.merge = function merge(other$0) {
  var __at___$7 = undefined;var self$4 = undefined;__at___$7 = this;self$4 = this;return __at___$7.nTracks = __at___$7.nTracks + other$0.nTracks;
};Counter$0.prototype.empty = function empty() {
  var __at___$8 = undefined;var self$5 = undefined;__at___$8 = this;self$5 = this;return __at___$8.nTracks <= 0;
};Counter$0.prototype.notEmpty = function () {
  var __at___$9 = undefined;var self$6 = undefined;__at___$9 = this;self$6 = this;return __at___$9.nTracks > 0;
};Counter$0.prototype.size = function size() {
  var __at___$10 = undefined;var self$7 = undefined;__at___$10 = this;self$7 = this;return 1;
};__amp____colon__(Counter$0, __amp____colon__(($targ$7 = "Counter", accum$4 = {}, accum$4["::name"] = $targ$7, accum$4), ($targ$8 = true, accum$5 = {}, accum$5["::egclass"] = $targ$8, accum$5)));Chirurgies$0 = function Chirurgies() {
  var $targ$9 = undefined;var __at___$11 = undefined;if (!getChecker(Chirurgies$0)(this)) {
    __at___$11 = Object.create(Chirurgies$0.prototype);
  } else {
    __at___$11 = this;
  }$targ$9 = new Map();__at___$11.refs = $targ$9;return __at___$11;
};Chirurgies$0.prototype.mark = function mark(temp$3$0) {
  var t0$3 = undefined;var t1$3 = undefined;var g$0 = undefined;var c$0 = undefined;var __at___$12 = undefined;var self$8 = undefined;__at___$12 = this;self$8 = this;t0$3 = temp$3$0;if (Array.isArray(t0$3) && (t1$3 = t0$3.length, t1$3 === 2)) {
    g$0 = t0$3[0];c$0 = t0$3[1];
  } else {
    ___match_error(temp$3$0);
  }return __at___$12.refs.set(g$0, c$0);
};Chirurgies$0.prototype.mergeOne = function (counts$0, temp$4$0) {
  var t0$4 = undefined;var t1$4 = undefined;var g$1 = undefined;var c$1 = undefined;var __at___$13 = undefined;var self$9 = undefined;__at___$13 = this;self$9 = this;t0$4 = temp$4$0;if (Array.isArray(t0$4) && (t1$4 = t0$4.length, t1$4 === 2)) {
    g$1 = t0$4[0];c$1 = t0$4[1];
  } else {
    ___match_error(temp$4$0);
  }if (counts$0.nTracks > 0) {
    return __at___$13.refs.set(g$1, c$1);
  }
};Chirurgies$0.prototype.merge = function merge(other$1) {
  var m$4 = undefined;var acc$3 = undefined;var temp$5 = undefined;var __at___$14 = undefined;var self$10 = undefined;__at___$14 = this;self$10 = this;acc$3 = [];temp$5 = null;m$4 = null;$3: for (var _iterator = other$1.refs.entries()[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$4 = _step.value;
    var g$2 = undefined;var c$2 = undefined;var t0$5 = undefined;var t1$5 = undefined;t0$5 = m$4;if (Array.isArray(t0$5) && (t1$5 = t0$5.length, t1$5 === 2)) {
      g$2 = t0$5[0];c$2 = t0$5[1];temp$5 = __at___$14.refs.set(g$2, c$2);acc$3.push(temp$5);
    } else {
      ___match_error(m$4);
    }
  }return acc$3;
};Chirurgies$0.prototype.unmergeOne = function (counts$1, temp$6$0) {
  var t0$6 = undefined;var t1$6 = undefined;var g$3 = undefined;var __at___$15 = undefined;var self$11 = undefined;__at___$15 = this;self$11 = this;t0$6 = temp$6$0;if (Array.isArray(t0$6) && (t1$6 = t0$6.length, t1$6 === 2)) {
    g$3 = t0$6[0];t0$6[1];
  } else {
    ___match_error(temp$6$0);
  }return __at___$15.refs["delete"](g$3);
};Chirurgies$0.prototype.unmerge = function unmerge(other$2) {
  var m$5 = undefined;var acc$4 = undefined;var temp$7 = undefined;var __at___$16 = undefined;var self$12 = undefined;__at___$16 = this;self$12 = this;acc$4 = [];temp$7 = null;m$5 = null;$4: for (var _iterator = other$2.refs.entries()[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$5 = _step.value;
    var g$4 = undefined;var t0$7 = undefined;var t1$7 = undefined;t0$7 = m$5;if (Array.isArray(t0$7) && (t1$7 = t0$7.length, t1$7 === 2)) {
      g$4 = t0$7[0];t0$7[1];temp$7 = __at___$16.refs["delete"](g$4);acc$4.push(temp$7);
    } else {
      ___match_error(m$5);
    }
  }return acc$4;
};Chirurgies$0.prototype.unmergeCounts = function (other$3) {
  var $targ$10 = undefined;var __at___$17 = undefined;var self$13 = undefined;__at___$17 = this;self$13 = this;if (other$3.nTracks > 0) {
    $targ$10 = new Map();__at___$17.refs = $targ$10;return void 0;
  }
};Chirurgies$0.prototype.empty = function empty() {
  var __at___$18 = undefined;var self$14 = undefined;__at___$18 = this;self$14 = this;return __at___$18.refs.size === 0;
};Chirurgies$0.prototype.notEmpty = function () {
  var __at___$19 = undefined;var self$15 = undefined;__at___$19 = this;self$15 = this;return __at___$19.refs.size > 0;
};Chirurgies$0.prototype.size = function size() {
  var __at___$20 = undefined;var self$16 = undefined;__at___$20 = this;self$16 = this;return __at___$20.refs.size;
};__amp____colon__(Chirurgies$0, __amp____colon__(($targ$11 = "Chirurgies", accum$6 = {}, accum$6["::name"] = $targ$11, accum$6), ($targ$12 = true, accum$7 = {}, accum$7["::egclass"] = $targ$12, accum$7)));Tracks$0 = function Tracks() {
  var $targ$13 = undefined;var t0$8 = undefined;var m$6$0 = undefined;var __at___$21 = undefined;if (!getChecker(Tracks$0)(this)) {
    __at___$21 = Object.create(Tracks$0.prototype);
  } else {
    __at___$21 = this;
  }m$6$0 = arguments;t0$8 = m$6$0.length;if (t0$8 >= 0 && t0$8 <= 1) {
    __at___$21.factory = 0 >= t0$8 ? Counter$0 : m$6$0[0];$targ$13 = {};__at___$21.data = $targ$13;void 0;
  } else {
    ___match_error(m$6$0, "{@factory = Counter}");
  }return __at___$21;
};Tracks$0.prototype._data = function _data() {
  var m$7 = undefined;var acc$5 = undefined;var temp$8 = undefined;var __at___$22 = undefined;var self$17 = undefined;__at___$22 = this;self$17 = this;return object((acc$5 = [], temp$8 = null, m$7 = null, (function () {
    $5: for (var _iterator = items(__at___$22.data)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      (function () {
        m$7 = _step.value;
        var m$8 = undefined;var acc$6 = undefined;var temp$9 = undefined;var id$0 = undefined;var tr$0 = undefined;var t0$9 = undefined;var t1$8 = undefined;t0$9 = m$7;if (Array.isArray(t0$9) && (t1$8 = t0$9.length, t1$8 === 2)) {
          id$0 = t0$9[0];tr$0 = t0$9[1];temp$8 = [id$0, object((acc$6 = [], temp$9 = null, m$8 = null, (function () {
            $6: for (var _iterator2 = items(tr$0)[Symbol.iterator](), _step2; !(_step2 = _iterator2.next()).done;) {
              m$8 = _step2.value;
              var k$0 = undefined;var v$2 = undefined;var t0$10 = undefined;var t1$9 = undefined;t0$10 = m$8;if (Array.isArray(t0$10) && (t1$9 = t0$10.length, t1$9 === 2)) {
                k$0 = t0$10[0];v$2 = t0$10[1];temp$9 = [k$0, v$2.notEmpty()];acc$6.push(temp$9);
              } else {
                ___match_error(m$8);
              }
            }
          })(), acc$6))];acc$5.push(temp$8);
        } else {
          ___match_error(m$7);
        }
      })();
    }
  })(), acc$5));
};Tracks$0.prototype.get = function get(id$1) {
  var __at___$23 = undefined;var self$18 = undefined;__at___$23 = this;self$18 = this;return id$1 && send(send(__at___$23, "data", true), id$1);
};Tracks$0.prototype.getFor = function (obj$0) {
  var id$2 = undefined;var __at___$24 = undefined;var self$19 = undefined;__at___$24 = this;self$19 = this;if (!obj$0) {
    return null;
  }id$2 = send(obj$0, ID$0);return id$2 && send(send(__at___$24, "data", true), id$2);
};Tracks$0.prototype.getForInit = function (obj$1, initializer$0) {
  var id$3 = undefined;var __at___$25 = undefined;var self$20 = undefined;__at___$25 = this;self$20 = this;if (!obj$1) {
    return null;
  }id$3 = send(obj$1, ID$0);return id$3 && (__at___$25.data[id$3] = send(send(__at___$25, "data", true), id$3) || initializer$0());
};Tracks$0.prototype.list = function list() {
  var __at___$26 = undefined;var self$21 = undefined;__at___$26 = this;self$21 = this;return items(__at___$26.data);
};Tracks$0.prototype.empty = function empty() {
  var __at___$27 = undefined;var self$22 = undefined;__at___$27 = this;self$22 = this;return keys(__at___$27.data).length === 0;
};Tracks$0.prototype.mark = function mark(obj$2, key$0, initializer$1) {
  var tracks$0 = undefined;var t$0 = undefined;var __at___$28 = undefined;var self$23 = undefined;__at___$28 = this;self$23 = this;if (!obj$2) {
    return null;
  }tracks$0 = __at___$28.getForInit(obj$2, initializer$1);t$0 = tracks$0[key$0] = send(tracks$0, key$0) || __at___$28.factory();return t$0.mark();
};Tracks$0.prototype.unmark = function unmark(obj$3, key$1) {
  var tracks$1 = undefined;var t$1 = undefined;var __at___$29 = undefined;var self$24 = undefined;__at___$29 = this;self$24 = this;tracks$1 = __at___$29.getFor(obj$3);if (!tracks$1) {
    return null;
  }t$1 = send(tracks$1, key$1);t$1.unmark();if (t$1.empty()) {
    delete tracks$1[key$1];
  }
};Tracks$0.prototype.merge = function merge() {
  var m$10 = undefined;var acc$7 = undefined;var temp$10 = undefined;var tracks$2 = undefined;var merger$0 = undefined;var arg$0 = undefined;var t0$11 = undefined;var m$9$0 = undefined;var __at___$30 = undefined;var self$25 = undefined;__at___$30 = this;self$25 = this;m$9$0 = arguments;t0$11 = m$9$0.length;if (t0$11 >= 1 && t0$11 <= 3) {
    tracks$2 = m$9$0[0];if (1 >= t0$11) {
      merger$0 = "merge";
    } else {
      merger$0 = m$9$0[1];
    }if (2 >= t0$11) {
      arg$0 = null;
    } else {
      arg$0 = m$9$0[2];
    }acc$7 = [];temp$10 = null;m$10 = null;$7: for (var _iterator = items(tracks$2.data)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
      m$10 = _step.value;
      var m$11 = undefined;var v0$0 = undefined;var id$4 = undefined;var v$3 = undefined;var t0$12 = undefined;var t1$10 = undefined;t0$12 = m$10;if (Array.isArray(t0$12) && (t1$10 = t0$12.length, t1$10 === 2)) {
        id$4 = t0$12[0];v$3 = t0$12[1];v0$0 = __at___$30.data[id$4] = send(send(__at___$30, "data", true), id$4) || Object.create(Object.getPrototypeOf(v$3));m$11 = null;$8: for (var _iterator2 = items(v$3)[Symbol.iterator](), _step2; !(_step2 = _iterator2.next()).done;) {
          m$11 = _step2.value;
          var obj0$0 = undefined;var key$2 = undefined;var obj$4 = undefined;var t0$13 = undefined;var t1$11 = undefined;t0$13 = m$11;if (Array.isArray(t0$13) && (t1$11 = t0$13.length, t1$11 === 2)) {
            key$2 = t0$13[0];obj$4 = t0$13[1];obj0$0 = v0$0[key$2] = send(v0$0, key$2) || __at___$30.factory();send(obj0$0, merger$0, true)(obj$4, arg$0);if (obj0$0.empty()) {
              delete v0$0[key$2];
            }
          } else {
            ___match_error(m$11);
          }
        }if (keys(v0$0).length === 0) {
          delete __at___$30.data[id$4];
        }acc$7.push(temp$10);
      } else {
        ___match_error(m$10);
      }
    }return acc$7;
  } else {
    return ___match_error(m$9$0, "{tracks, merger = .merge, arg = null}");
  }
};Tracks$0.prototype.size = function size() {
  var m$12 = undefined;var sz$0 = undefined;var __at___$31 = undefined;var self$26 = undefined;__at___$31 = this;self$26 = this;sz$0 = 0;m$12 = null;$9: for (var _iterator = items(__at___$31.data)[Symbol.iterator](), _step; !(_step = _iterator.next()).done;) {
    m$12 = _step.value;
    var m$13 = undefined;var acc$8 = undefined;var temp$11 = undefined;var id$5 = undefined;var ob$0 = undefined;var t0$14 = undefined;var t1$12 = undefined;t0$14 = m$12;if (Array.isArray(t0$14) && (t1$12 = t0$14.length, t1$12 === 2)) {
      id$5 = t0$14[0];ob$0 = t0$14[1];acc$8 = [];temp$11 = null;m$13 = null;$10: for (var _iterator2 = items(ob$0)[Symbol.iterator](), _step2; !(_step2 = _iterator2.next()).done;) {
        m$13 = _step2.value;
        var k$1 = undefined;var v$4 = undefined;var t0$15 = undefined;var t1$13 = undefined;t0$15 = m$13;if (Array.isArray(t0$15) && (t1$13 = t0$15.length, t1$13 === 2)) {
          k$1 = t0$15[0];v$4 = t0$15[1];temp$11 = sz$0 = sz$0 + v$4.size();acc$8.push(temp$11);
        } else {
          ___match_error(m$13);
        }
      }acc$8;
    } else {
      ___match_error(m$12);
    }
  }return sz$0;
};__amp____colon__(Tracks$0, __amp____colon__(($targ$14 = "Tracks", accum$8 = {}, accum$8["::name"] = $targ$14, accum$8), ($targ$15 = true, accum$9 = {}, accum$9["::egclass"] = $targ$15, accum$9)));Patch$0 = function Patch() {
  var t0$16 = undefined;var m$14$0 = undefined;var __at___$32 = undefined;if (!getChecker(Patch$0)(this)) {
    __at___$32 = Object.create(Patch$0.prototype);
  } else {
    __at___$32 = this;
  }m$14$0 = arguments;t0$16 = m$14$0.length;if (t0$16 >= 0 && t0$16 <= 1) {
    __at___$32.data = 0 >= t0$16 ? {} : m$14$0[0];undefined;
  } else {
    ___match_error(m$14$0, "{@data = {=}}");
  }return __at___$32;
};Patch$0.prototype._data = function _data() {
  var __at___$33 = undefined;var self$27 = undefined;__at___$33 = this;self$27 = this;return __at___$33.data;
};Patch$0.prototype.get = function get(id$6) {
  var __at___$34 = undefined;var self$28 = undefined;__at___$34 = this;self$28 = this;return __at___$34.data[id$6] = send(send(__at___$34, "data", true), id$6) || {};
};Patch$0.prototype.getFor = function (obj$5) {
  var id$7 = undefined;var __at___$35 = undefined;var self$29 = undefined;__at___$35 = this;self$29 = this;if (!obj$5) {
    return null;
  }id$7 = send(obj$5, ID$0);return id$7 && send(send(__at___$35, "data", true), id$7);
};Patch$0.prototype.getForInit = function (obj$6, initializer$2) {
  var id$8 = undefined;var __at___$36 = undefined;var self$30 = undefined;__at___$36 = this;self$30 = this;if (!obj$6) {
    return null;
  }id$8 = send(obj$6, ID$0);return id$8 && (__at___$36.data[id$8] = send(send(__at___$36, "data", true), id$8) || initializer$2());
};Patch$0.prototype.register = function register(obj$7, key$3, val$0, initializer$3) {
  var $targ$16 = undefined;var patch$0 = undefined;var __at___$37 = undefined;var self$31 = undefined;__at___$37 = this;self$31 = this;patch$0 = __at___$37.getForInit(obj$7, initializer$3);if (patch$0) {
    $targ$16 = val$0;patch$0[key$3] = $targ$16;return void 0;
  }
};Patch$0.prototype.shift = function shift() {
  var t0$18 = undefined;var t1$14 = undefined;var $targ$17 = undefined;var $get$0$0 = undefined;var new__minus__data$0 = undefined;var t0$17 = undefined;var m$15$0 = undefined;var __at___$38 = undefined;var self$32 = undefined;__at___$38 = this;self$32 = this;m$15$0 = arguments;t0$17 = m$15$0.length;if (t0$17 >= 0 && t0$17 <= 1) {
    if (0 >= t0$17) {
      new__minus__data$0 = {};
    } else {
      new__minus__data$0 = m$15$0[0];
    }$targ$17 = [__at___$38.data, new__minus__data$0];t0$18 = $targ$17;if (Array.isArray(t0$18) && (t1$14 = t0$18.length, t1$14 === 2)) {
      $get$0$0 = t0$18[0];__at___$38.data = t0$18[1];
    } else {
      ___match_error($targ$17, "{expr-value, @data}");
    }return $get$0$0;
  } else {
    return ___match_error(m$15$0, "{new-data = {=}}");
  }
};Patch$0.prototype.list = function list() {
  var __at___$39 = undefined;var self$33 = undefined;__at___$39 = this;self$33 = this;return items(__at___$39.data);
};Patch$0.prototype.empty = function empty() {
  var __at___$40 = undefined;var self$34 = undefined;__at___$40 = this;self$34 = this;return keys(__at___$40.data).length === 0;
};__amp____colon__(Patch$0, __amp____colon__(($targ$18 = "Patch", accum$10 = {}, accum$10["::name"] = $targ$18, accum$10), ($targ$19 = true, accum$11 = {}, accum$11["::egclass"] = $targ$19, accum$11)));$targ$20 = ObjectTrack$0;exports.ObjectTrack = $targ$20;$targ$21 = ArrayTrack$0;exports.ArrayTrack = $targ$21;$targ$22 = Counter$0;exports.Counter = $targ$22;$targ$23 = Chirurgies$0;exports.Chirurgies = $targ$23;$targ$24 = Tracks$0;exports.Tracks = $targ$24;$targ$25 = Patch$0;exports.Patch = $targ$25;void 0;
